---
title: "Supporting information"
author: "Tom Ellis"
date: "`r format(Sys.time(), '%d %B %Y')`"
output:
  bookdown::pdf_book:
    toc: no
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, fig.pos = 'H')
library("qtl", quietly = T)
library(qtltools)
```

This Rmarkdown file contains the files necessary to recreate the results presented in the manuscript "Positive correlations among fecundity components boost estimates of local adaptation in *Arabidopsis thaliana*" by Tom Ellis and Jon Ã…gren. If you are reading a PDF, HTML or MS Word version of this document, see the original `.Rmd` file for the full R source code. See also the file `main_results.Rmd` for code to generate the data in the tables.

Summary of the genotype data in the R/QTL cross files.

```{r import-qtl-data, include=FALSE}
# Import R/qtl data on seed number, mass and total estimated fitness (seed number * total fruit number).
library("qtl", quietly = T)

# There are QTL models for each of four site-year combinations
siteyear <- c("it2010", "it2011", "sw2010", "sw2011")
# There are six phenotypes in each site-year combination:
traits <- c(
  "mass", # 1. seed mass (mass)
  "frut", # 2. fruits/plant ()
  "seed", # 3. seeds/fruit
  "tofu", # 4. Total fecundity (estimated seeds/plant = frut*seed)
  "ffit", # 5. Fruits per planted seedling; fitness measure used by Agren etal 2013
  "tfit", # 6. Seeds per planted seedling; fitness incorporating seed number.
  "surv") # 7. Survival
# parameters to input
ix <- traits[7] # pull out the name of the trait we are mapping here

# The following loop imports cross and qtl objects, then fits ANOVA models and clusters into groups.
# first, set up empty lists to hold this information.
cross <-              qtlmodels  <-       qtlfits  <- lapply(traits, function(x) return(NULL))
names(cross) <- names(qtlmodels) <- names(qtlfits) <- traits

for(i in traits){
  # Import rqtl cross objects.
  thiscross <- read.cross("csv", "", paste("data_files/rqtl_", i,".csv",sep=""),
                          na.strings=NA, genotypes = c("a", "b"), alleles=c("It","Sw"))
  thiscross <- convert2riself(thiscross)
  thiscross <- calc.genoprob(thiscross, step=2, error.prob=.0001, map.function="kosambi",stepwidth="max", off.end=0)
  cross[[i]] <- thiscross
  
  # import objects containing stepwise QTL fitting results.
  thisqtl <- readRDS(paste('output/stepwise_',i,'.rds', sep=""))
  qtlmodels[[i]] <- thisqtl
  
  # fit ANOVAs to each QTL model
  thistrait <- lapply(siteyear, function(x) NULL)
  names(thistrait) <- siteyear
  # loop over individual years and get model fits for each
  for(y in siteyear){
    # For survival in Sweden 2010, no QTL were identified. This if statement checks for that.
    if(length(thisqtl[[y]]) == 0 ){
      thistrait[[y]] <- NA
    } else {
      thistrait[[y]] <- fitqtl(cross = thiscross, pheno.col = y, qtl = thisqtl[[y]], formula = formula(thisqtl[[y]]),
                               method = "hk", model = "normal", covar = NULL, get.ests = T, dropone = T)}
    qtlfits[[i]] <- thistrait
  }
}

# remove the entry for Survival in Sweden 2010 from qtlmodels
# qtlmodels$surv <- qtlmodels$surv[c("it2010", "it2011", "sw2011")]

```

```{r cluster-qtl, include=F}
qtlclusters  <- lapply(traits, function(x) return(NULL))
names(qtlclusters) <- traits

# perform QTL clustering for each trait
library(qtltools)
for(i in traits){
  if(i == "surv"){
    dat <- cluster_qtl(1:5, qtlmodels[[i]][c(1,2,4)], qtlfits[[i]][c(1,2,4)], 15.2)$full.list
  } else {
    dat <- cluster_qtl(1:5, qtlmodels[[i]], qtlfits[[i]], 15.2)$full.list
  }
  qtlclusters[[i]] <- dat
  
}

qtlclusters$surv$experiment[qtlclusters$surv$experiment == 3] <- 4
```

```{r}
dat <- qtlclusters$mass

dat
```

\newpage

```{r tab:individual-mass-qtl}
siteyear <- c("italy 2010", "italy 2011", "sweden 2010", "sweden 2011")

# summary tables for each experiment
mass_tab <- list(bayesint_table(mass_stepwise$it2010, mass_fit$it2010),
                 bayesint_table(mass_stepwise$it2011, mass_fit$it2011),
                 bayesint_table(mass_stepwise$sw2010, mass_fit$sw2010),
                 bayesint_table(mass_stepwise$sw2011, mass_fit$sw2011))
for(i in 1:4) mass_tab[[i]]$experiment <- rep(siteyear[i], nrow(mass_tab[[i]])) # add experiment names
mass_tab <- do.call('rbind', mass_tab)
# mash together positions with their CIs
mass_tab$position <- paste(round(mass_tab$ML_bayesint,1), " (", round(mass_tab$min_bayesint, 1), "-", round(mass_tab$max_bayesint, 1), ")", sep="")
# print the table
knitr::kable(mass_tab[,c("experiment","chr","position","LOD", "effect_sizes","PVE")], digits = c(0,0,0,1,3,1),
             col.names = c("experiment","chromosome","position (cM)","LOD", "effect size","% var. explained"),
             caption="QTL for seed mass identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals.")
```

\newpage

```{r tab:individual-frut-qtl}
# fruit per plant
frut_tab <- list(bayesint_table(frut_stepwise$it2010, frut_fit$it2010),
                 bayesint_table(frut_stepwise$it2011, frut_fit$it2011),
                 bayesint_table(frut_stepwise$sw2010, frut_fit$sw2010),
                 bayesint_table(frut_stepwise$sw2011, frut_fit$sw2011))
for(i in 1:4) frut_tab[[i]]$experiment <- rep(siteyear[i], nrow(frut_tab[[i]])) # add experiment names
frut_tab <- do.call('rbind', frut_tab)
# mash together positions with their CIs
frut_tab$position <- paste(round(frut_tab$ML_bayesint,1), " (", round(frut_tab$min_bayesint, 1), "-", round(frut_tab$max_bayesint, 1), ")", sep="")
# print the table
knitr::kable(frut_tab[,c("experiment","chr","position","LOD", "effect_sizes","PVE")], digits = c(0,0,0,1,3,1),
             col.names = c("experiment","chromosome","position (cM)","LOD", "effect size","% var. explained"),
             caption="QTL for fruit number per reproductive plant identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals.")
```

\newpage

```{r tab:individual-numb-qtl}
# seed number per fruit
seed_tab <- list(bayesint_table(seed_stepwise$it2010, seed_fit$it2010),
                 bayesint_table(seed_stepwise$it2011, seed_fit$it2011),
                 bayesint_table(seed_stepwise$sw2010, seed_fit$sw2010),
                 bayesint_table(seed_stepwise$sw2011, seed_fit$sw2011))
for(i in 1:4) seed_tab[[i]]$experiment <- rep(siteyear[i], nrow(seed_tab[[i]])) # add experiment names
seed_tab <- do.call('rbind', seed_tab)
# mash together positions with their CIs
seed_tab$position <- paste(round(seed_tab$ML_bayesint,1), " (", round(seed_tab$min_bayesint, 1), "-", round(seed_tab$max_bayesint, 1), ")", sep="")
# print the table
knitr::kable(seed_tab[,c("experiment","chr","position","LOD", "effect_sizes","PVE")], digits = c(0,0,0,1,3,1),
             col.names = c("experiment","chromosome","position (cM)","LOD", "effect size","% var. explained"),
             caption="QTL for seed number per fruit identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals.")
```

\newpage

```{r tab:individual-tofu-qtl}
# total fecundity
tofu_tab <- list(bayesint_table(tofu_stepwise$it2010, tofu_fit$it2010),
                 bayesint_table(tofu_stepwise$it2011, tofu_fit$it2011),
                 bayesint_table(tofu_stepwise$sw2010, tofu_fit$sw2010),
                 bayesint_table(tofu_stepwise$sw2011, tofu_fit$sw2011))
for(i in 1:4) tofu_tab[[i]]$experiment <- rep(siteyear[i], nrow(tofu_tab[[i]])) # add experiment names
tofu_tab <- do.call('rbind', tofu_tab)
# mash together positions with their CIs
tofu_tab$position <- paste(round(tofu_tab$ML_bayesint,1), " (", round(tofu_tab$min_bayesint, 1), "-", round(tofu_tab$max_bayesint, 1), ")", sep="")
# print the table
knitr::kable(tofu_tab[,c("experiment","chr","position","LOD", "effect_sizes","PVE")], digits = c(0,0,0,1,1,1),
             col.names = c("experiment","chromosome","position (cM)","LOD", "effect size","% var. explained"),
             caption = "QTL for seed number per reproductive plant identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals.")

```

```{r tab: epistatic-qtl}

lst <- list(mass_fit$sw2011, frut_fit$it2010, frut_fit$sw2011, 
     seed_fit$it2011, seed_fit$sw2010, seed_fit$sw2011, tofu_fit$it2010, tofu_fit$sw2011)

epi_fun <- function(x) {
  d <- summary(x)
  i <- nrow(d$result.drop)
  c(row.names(d$ests)[i+1], d$ests[i, 1], d$result.drop[i, c(3,4)])
}

epi_tab <- cbind(trait = c("seed mass", 
                           "fruits per reproductive adult", "fruits per reproductive adult", "fruits per reproductive adult",
                           "seeds per fruit", "seeds per fruit",
                           "seeds per reproductive adult", "seeds per reproductive adult"),
                 site = c("sweden", "italy", "sweden", "italy", "sweden", "sweden", "italy", "sweden"),
                 year = c(2011, 2010, 2011, 2011, 2010, 2011, 2010, 2011),
                 as.data.frame(t(sapply(lst, epi_fun)))
)
colnames(epi_tab) <- c("trait",   "site", "year", "loci", "effect size", "LOD", "% variance explained")
for(c in 5:7) epi_tab[,c] <- as.numeric(as.vector(epi_tab[,c]))

knitr::kable(epi_tab, digits = 2,
             caption = "Pairs of significant epistatic QTL.")
rm(lst)
```

\newpage

```{r fig:ind-years-mass, include=F, fig.width=21/2.54, fig.cap="Figure S1: QTL for seed mass in each year in Italy (red) and Sweden (blue). Arrows show the maximum-likelihood position of the QTL and bars indicate the extent of 95% Bayesian credible intervals. The direction of the arrow indicates the direction of effect of the Swedish allele on the phenotype. Open triangles are 'poorly defined' QTL with credible intervals greater than 15.2cM."}
mloc <- read.table('../data_files/genetic_map_agren_etal_2013.csv', sep = '\t', header=T)
# pdf('../figures/QTL_positions_frut.pdf', width=23/2.56, height = 18/2.56, family='Times')
par(mfrow=c(1,1), mar=c(0,3,0,0)+0.1, oma=c(1,0,0,0), family='Times')
# set up plot
mass_plot <- plotQTL(1:5, mloc, nlanes = 4, left_gap = 4, right_gap = 0)
plotQTL_base(mass_plot, by=20)
text(33, -92, "Chromosome")
# import fitness boxes and plot.
boxes <- read.csv("../data_files/fitness_boxes.csv")
boxes$track <- match(boxes$chr, colnames(mass_plot$lane_margins)) # vector denoting which track to plot to.
for(b in 1:nrow(boxes)) plotQTL_box(mass_plot, boxes$track[b], boxes$upper[b], boxes$lower[b])
# Lanes and labels
plotQTL_lanes(mass_plot)
text(mass_plot$lane_centres,     3, 2010:2011, col="gray50", cex=0.7, srt=90)
text(mass_plot$lane_margins[2,], 8, "Italy",   col='red3',   cex=0.8, adj = c(0.5,0))
text(mass_plot$lane_margins[4,], 8, "Sweden",  col='blue4',  cex=0.8, adj = c(0.5,0))
# plot QTL for each site-year combination
colvec <- rep(c('red3', 'blue4'),each=2)
for(y in 1:4){
  plotQTL_qtl(mass_plot, mass_stepwise[[y  ]], mass_fit[[y  ]], lane = y,   col = colvec[y],  fill_cutoff = 15.2, lwd=2)
}
```

```{r fig:ind-years-frut, include=F, fig.width=21/2.54, fig.cap="QTL for fruit number per reproductive plant in each year in Italy (red) and Sweden (blue). Arrows show the maximum-likelihood position of the QTL and bars indicate the extent of 95% Bayesian credible intervals. The direction of the arrow indicates the direction of effect of the Swedish allele on the phenotype. Open triangles are 'poorly defined' QTL with credible intervals greater than 15.2cM."}
library(qtl)
library(qtltools)
mloc <- read.table('../data_files/genetic_map_agren_etal_2013.csv', sep = '\t', header=T)

# pdf('../figures/QTL_positions_number.pdf', width=23/2.56, height = 18/2.56, family='Times')
par(mfrow=c(1,1), mar=c(0,3,0,0)+0.1, oma=c(1,0,0,0), family='Times')
# set up plot
frut_plot <- plotQTL(1:5, mloc, nlanes = 4, left_gap = 4, right_gap = 0)
plotQTL_base(frut_plot, by=20)
text(33, -92, "Chromosome")
# import fitness boxes and plot.
boxes <- read.csv("../data_files/fitness_boxes.csv")
boxes$track <- match(boxes$chr, colnames(frut_plot$lane_margins)) # vector denoting which track to plot to.
for(b in 1:nrow(boxes)) plotQTL_box(frut_plot, boxes$track[b], boxes$upper[b], boxes$lower[b])
# Lanes and labels
plotQTL_lanes(frut_plot)
# segments(frut_plot$lane_margins[6,], 0, frut_plot$lane_margins[6,], -frut_plot$track_lengths)
text(frut_plot$lane_centres,     3, 2010:2011, col="gray50", cex=0.7, srt=90)
text(frut_plot$lane_margins[2,], 8, "Italy",   col='red3',   cex=0.8, adj = c(0.5,0))
text(frut_plot$lane_margins[4,], 8, "Sweden",  col='blue4',  cex=0.8, adj = c(0.5,0))
# plot QTL for each site-year combination
colvec <- rep(c('red3', 'blue4'), each=2)
for(y in 1:4){
  plotQTL_qtl(frut_plot, frut_stepwise[[y  ]], frut_fit[[y  ]], lane = y,   col = colvec[y],  fill_cutoff = 15.2, lwd=2)
}
```

```{r fig:ind-years-seednumber, include=F, fig.width=21/2.54, fig.cap="QTL for seed number per plant in each year in Italy (red) and Sweden (blue). Arrows show the maximum-likelihood position of the QTL and bars indicate the extent of 95% Bayesian credible intervals. The direction of the arrow indicates the direction of effect of the Swedish allele on the phenotype. Open triangles are 'poorly defined' QTL with credible intervals greater than 15.2cM."}
mloc <- read.table('../data_files/genetic_map_agren_etal_2013.csv', sep = '\t', header=T)
# pdf('../figures/QTL_positions_tofu.pdf', width=23/2.56, height = 18/2.56, family='Times')
par(mfrow=c(1,1), mar=c(0,3,0,0)+0.1, oma=c(1,0,0,0), family='Times')
# set up plot
seed_plot <- plotQTL(1:5, mloc, nlanes = 4, left_gap = 4, right_gap = 0)
plotQTL_base(seed_plot, by=20)
text(33, -92, "Chromosome")
# import fitness boxes and plot.
boxes <- read.csv("../data_files/fitness_boxes.csv")
boxes$track <- match(boxes$chr, colnames(seed_plot$lane_margins)) # vector denoting which track to plot to.
for(b in 1:nrow(boxes)) plotQTL_box(seed_plot, boxes$track[b], boxes$upper[b], boxes$lower[b])
# Lanes and labels
plotQTL_lanes(seed_plot)
# segments(seed_plot$lane_margins[6,], 0, seed_plot$lane_margins[6,], -seed_plot$track_lengths)
text(seed_plot$lane_centres,     3, 2010:2011, col="gray50", cex=0.7, srt=90)
text(seed_plot$lane_margins[2,], 8, "Italy",   col='red3',   cex=0.8, adj = c(0.5,0))
text(seed_plot$lane_margins[4,], 8, "Sweden",  col='blue4',  cex=0.8, adj = c(0.5,0))
# plot QTL for each site-year combination
colvec <- rep(c('red3', 'blue4'), each=2)
for(y in 1:4){
  plotQTL_qtl(seed_plot, seed_stepwise[[y  ]], seed_fit[[y  ]], lane = y,   col = colvec[y],  fill_cutoff = 15.2, lwd=2)
}

```

```{r fig:ind-years-fecundity, include=F, fig.width=21/2.54, fig.cap="QTL for seed number per reproductive plant in each year in Italy (red) and Sweden (blue). Arrows show the maximum-likelihood position of the QTL and bars indicate the extent of 95% Bayesian credible intervals. The direction of the arrow indicates the direction of effect of the Swedish allele on the phenotype. Open triangles are 'poorly defined' QTL with credible intervals greater than 15.2cM."}
mloc <- read.table('../data_files/genetic_map_agren_etal_2013.csv', sep = '\t', header=T)
# pdf('../figures/QTL_positions_tofu.pdf', width=23/2.56, height = 18/2.56, family='Times')
par(mfrow=c(1,1), mar=c(0,3,0,0)+0.1, oma=c(1,0,0,0), family='Times')
# set up plot
tofu_plot <- plotQTL(1:5, mloc, nlanes = 4, left_gap = 4, right_gap = 0)
plotQTL_base(tofu_plot, by=20)
text(33, -92, "Chromosome")
# import fitness boxes and plot.
boxes <- read.csv("../data_files/fitness_boxes.csv")
boxes$track <- match(boxes$chr, colnames(tofu_plot$lane_margins)) # vector denoting which track to plot to.
for(b in 1:nrow(boxes)) plotQTL_box(tofu_plot, boxes$track[b], boxes$upper[b], boxes$lower[b])
# Lanes and labels
plotQTL_lanes(tofu_plot)
# segments(tofu_plot$lane_margins[6,], 0, tofu_plot$lane_margins[6,], -tofu_plot$track_lengths)
text(tofu_plot$lane_centres,     3, 2010:2011, col="gray50", cex=0.7, srt=90)
text(tofu_plot$lane_margins[2,], 8, "Italy",   col='red3',   cex=0.8, adj = c(0.5,0))
text(tofu_plot$lane_margins[4,], 8, "Sweden",  col='blue4',  cex=0.8, adj = c(0.5,0))
# plot QTL for each site-year combination
colvec <- rep(c('red3', 'blue4'), each=2)
for(y in 1:4){
  plotQTL_qtl(tofu_plot, tofu_stepwise[[y  ]], tofu_fit[[y  ]], lane = y,   col = colvec[y],  fill_cutoff = 15.2, lwd=2)
}

```

```{r fig: epistatic-mass, fig.width=8/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for the epistatic interaction detected for seed mass, showing mean and standard errors for each genotype x genotype combination."}
# seed mass
par(mfrow=c(1,1))
effectplot(massqtl, pheno.col = "mean_sw2010", mname1 = "2@22.6", mname2 = "5@2.6", add.legend = F,
           main = "Sweden 2010", xlab="QTL 5:2.6", ylab="seed mass (mg)")
legend(0.9,26.8, c("QTL 3:65.9","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")
```

```{r fig: epistatic-seed, fig.width=16.9/2.54, fig.height=10/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for epistatic interactions detected for seed number per fruit, showing mean and standard errors for each genotype x genotype combination."}
# seed number per fruit
par(mfrow=c(1,3))
effectplot(seedqtl, pheno.col = "mean_it2011", mname1 = "1@5.3", mname2 = "5@15.0", add.legend = F,
           main = "Italy 2011", xlab="QTL 5:15.0", ylab="seed number per fruit")
legend(1.4,30.5, c("QTL 1:5.3","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")

effectplot(seedqtl, pheno.col = "mean_sw2010", mname1 = "1@22.7", mname2 = "5@70.1", add.legend = F,
           main = "Sweden 2010", xlab="QTL 5:70.1", ylab="seed number per fruit")
legend(1.4,33, c("QTL 1:22.7","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")

effectplot(seedqtl, pheno.col = "mean_sw2011", mname1 = "2@22.6", mname2 = "5@2.6", add.legend = F,
           main = "Sweden 2011", xlab="QTL 5:2.6", ylab="seed number per fruit")
legend(1.4,29.7, c("QTL 2:22.6","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")
```

```{r fig: epistatic-frut, fig.width=16.9/2.54, fig.height=10/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for epistatic interactions detected for fruit number per reproductive plant, showing mean and standard errors for each genotype x genotype combination."}
#fruit number per plant
par(mfrow=c(1,2))
effectplot(frutqtl, pheno.col = "mean_it2010", mname1 = "5@7.7", mname2 = "5@70.1", add.legend = F,
           main = "Italy 2010", xlab="QTL 5:70.1", ylab="fruit number per reproductive plant")
legend(1.4,16, c("QTL 5:7.7","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")
effectplot(frutqtl, pheno.col = "mean_sw2011", mname1 = "1@61.8", mname2 = "3@26.7", add.legend = F, ylim=c(16,30),
           main = "Sweden 2011", xlab="QTL 3:26.7", ylab="fruit number per reproductive plant")
legend(0.8,21, c("QTL 1:61.8","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")
```

```{r fig: epistatic-tofu, fig.width=16.9/2.54, fig.height=10/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for epistatic interactions detected for seed number per reproductive plant, showing mean and standard errors for each genotype x genotype combination."}
# seed number per plant
par(mfrow=c(1,2))
effectplot(tofuqtl, pheno.col = "mean_it2010", mname1 = "3@6.3", mname2 = "5@12.0", add.legend = F,
           main = "Italy 2010", xlab="QTL 5:12.0", ylab="seed number per reproductive plant")
legend(1.4,450, c("QTL 3:6.3","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")

effectplot(tofuqtl, pheno.col = "mean_sw2011", mname1 = "1@61.1", mname2 = "3@24.4", add.legend = F, ylim=c(500,1100),
           main = "Sweden 2011", xlab="QTL 3:24.4", ylab="seed number per reproductive plant")
legend(0.8,700, c("QTL 1.61.1","ItIt", "SwSw"), col=c(0,2,4), pch=1, bty="n")

```
