---
title: Life-history trade-offs and the genetic basis of fitness in *Arabidopsis thaliana*
# author: "Thomas James Ellis, Froukje M. Postma, Christopher G. Oakley and Jon Ågren"
# date: "`r format(Sys.time(), '%d.%m.%Y')`"
output:
  bookdown::pdf_book:
    toc: no
    keep_tex: yes
  bookdown::word_document2: null
always_allow_html: yes
bibliography: ../../../bibtex_files/tellis.bib
documentclass: article
# fontfamily: mathpazo
# fontsize: 12pt
# geometry: margin=1in
header-includes:
  \usepackage{setspace}\doublespacing
  \usepackage{lineno}
  \linenumbers
  \usepackage{float} \floatplacement{figure}{H} 
  \newcommand{\beginsupplement}{\setcounter{table}{0}  \renewcommand{\thetable}{S\arabic{table}} \setcounter{figure}{0} \renewcommand{\thefigure}{S\arabic{figure}}}
# csl: ../../bibtex_files/mol-ecol.csl
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
Sys.setenv(LANG = "en")
Sys.time()
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, cache=F, cache.lazy=F)

library('knitr')
library('qtl')
library('arghqtl')

```

```{r data-import, include=F}
# Formatting of parental and wilcox tests are done in separate scripts.
source("R/001.parents.R")
source("R/005.mass_wilcox_tests.R")

# Confidence intervals on seletcion coefficients
sel_boots <- readRDS("output/bootstrap_selection_coefficients.rds")
sel_cis <- lapply(sel_boots, function(x) apply(x, 2, quantile, c(0.025, 0.975)))

# FITNESS DIFFERENCES AMONG PARENTS
maxvals <- apply(parents[,c(4,7)], 1, max) # get the fittest phenotype in each site × year combination.
# relative fitnesses of the Italian and Swedish parents.
parents$it_relw <- parents$it_parent / maxvals
parents$sw_relw <- parents$sw_parent / maxvals
# selective advantage of the local morph
parents$s[parents$site == 'italy']  <- parents$it_relw[parents$site == 'italy']  - parents$sw_relw[parents$site == 'italy']
parents$s[parents$site == 'sweden'] <- parents$sw_relw[parents$site == 'sweden'] - parents$it_relw[parents$site == 'sweden']

# RIL phenotypes
ril_numb <- read.csv('data_derived/RIL_seeds_per_fruit.csv')
ril_mass <- read.csv('data_derived/RIL_seed_mass.csv')
ril_frut <- read.csv("data_derived/RIL_fruits_per_adult.csv")
ril_tofu <- read.csv('data_derived/RIL_seeds_per_adult.csv')
ril_surv <- read.csv('data_derived/RIL_survival.csv')

# Heritability estimates and confidence intervals
H2 <- readRDS("output/heritabilities.rds")
# Genetic correlations
correlations <- read.csv('output/genetic_correlations.csv')

# import objects from QTL mapping. See 'R/013.format_QTL_models.R' and the scripts that calls for the full analysis details.
qtl_stepwise <- readRDS(file='output/qtl_stepwise_models.rds')
qtl_fits     <- readRDS(file='output/qtl_model_fits.rds')
qtl_clusters <- readRDS(file="output/clusters_all_models.rds")

# Marker positions for plotting mapping results
mloc <- read.table('data_raw/genetic_map_agren_etal_2013.csv', sep = '\t', header=T)

site_names  <- c("Italy", "Italy", "Sweden", "Sweden")
year_names  <- c(2010, 2011, 2010, 2011)
```

**Authors**: Thomas James Ellis^1,2^, Froukje M. Postma^1^, Christopher G. Oakley^3^ and Jon Ågren^1^

**Affiliations**:

1. Plant Ecology and Evolution, Dept. of Ecology and Genetics, EBC, Uppsala University, Norbyvägen 18D, 75236 Uppsala, Sweden
2. Gregor Mendel Insitute of Molecular Plant Sciences, Doktor-Bohr-Gasse 3, 1030 Vienna, Austria
3. Department of Botany and Plant Pathology & the Center for Plant Biology, Purdue University, 915 W. State Street, West Lafayette, IN 47907-2054, United States

**Keywords**: local adaptation, pleiotropy, trade-off, reciprocal transplant, fitness components

\newpage

# Abstract

Local adaptation may entail trade-offs in resource allocation to survival and reproduction, but also an increase in overall availability of resources (“condition”) through improved capacity for coping with local environmental factors. We examined the relative contribution of trade-offs and increased condition to adaptive evolution in a recombinant-inbred-line population of *Arabidopsis thaliana* planted at the native sites of the parental ecotypes in Italy and Sweden in two years. We estimated genetic correlations among fitness components based on genotypic means, and explored their causes with QTL mapping. Local ecotypes produced more seeds per fruit than did non-local ecotypes, reflected in stronger adaptive differentiation than was previously shown based on survival and fruit number only. Genetic correlations between survival and overall fecundity, and between number of fruits and number of seeds per fruit, were positive, and there was little evidence of a trade-off between seed size and number. Quantitative trait loci for these traits tended to map to the same regions of the genome, and showed positive pleiotropic effects. The results indicate that adaptive differentiation between the two focal populations largely reflects the evolution of increased ability to acquire resources in the local environment, rather than shifts in the relative allocation to different life-history traits. Differentiation both in phenology and tolerance to cold are likely to contribute to the advantage of the local genotype at the two sites.
# Introduction

Adaptation to the local environment is reflected in an increase in mean population fitness in response to local selection pressures [@Williams1966; @Kawecki2004]. This may be associated with a shift in relative allocation of resources to optimise the balance between components of fitness, such as between reproduction and survival, or offspring size and number [@Williams1966; @Williams1966a; @Smith1974; @schluter1991conflicting]. Adaptation may also be associated with the evolution of traits increasing the amount of resources available to individuals (their “condition”) in the local environment, which could supersede the effects of variation in relative allocation to different functions [@VanNoordwijk1986; @schluter1991conflicting]. Condition is likely to be a complex phenotype affected by many traits, including those influencing the ability to tolerate local environmental stresses, and those influencing the match between the timing of life-history transitions and seasonal changes in resource availability. Variation in condition versus allocation have distinct implications for local adaptation. With variation in allocation, trade-offs place a constraint on adaptive evolution, and will be reflected as negative correlations between components of fitness. By contrast, variation in condition should allow for increased allocation to multiple components of fitness and thus positive correlations between fitness components. A full understanding of local adaptation and life-history evolution therefore requires knowledge of the genetic basis of overall fitness and the genetic basis of correlations between different components of fitness.

If fitness components are correlated, this raises two questions about their genetic architecture. First, to what extent are fitness components affected by loci with pleiotropic effects on multiple components? Second, if loci do show pleiotropy, are the directions of allelic effects on affected traits consistent across all loci and with the signs of correlations among components of fitness? If there are genetic trade-offs among fitness components, we expect to observe antagonistic pleiotropic effects of individual quantitative trait loci (QTL), whereby an allele is associated with an increase in one component of fitness, but a decrease in one or more other components [@Hazel1943; @Falconer1996]. On the other hand, if variation in condition is large we expect positive pleiotropy, whereby alleles at QTL affecting resource status are associated with changes in two or more fitness components in the same direction, causing phenotypes to be positively correlated [@Houle1991]. In reality, both processes are likely to be acting, and it is the relative strength of trade-offs and variation in condition that will determine whether negative or positive correlations are observed.

In plants, three trade-off relationships are likely to be especially relevant for overall fitness. Firstly, resources invested in reproduction are not available for growth and defence, causing a trade-off between fecundity and survival. A trade-off between reproduction and subsequent survival has been documented in many iteroparous species [@Williams1966a; @edward2011mechanisms]. However, a trade-off between reproduction and survival can also be expected in semelparous organisms if traits increasing the chance of juvenile survival reduces resources available to reproduction. Thus, we expect a negative correlation between fecundity and survival.

Second, total seed production is a function of both the number of fruits produced and the number of seeds per fruit, and there may be a trade-off between these two components of fecundity. For practical reasons, studies of local adaptation in plants typically focus on either fruit production or estimates of total seed production as a measure of fecundity [e.g. @Latta2009; @Hall2010; @fournier2011map; @hancock2011adaptation; @agren_genetic_2013]. To quantify components of fecundity, it is necessary to estimate both number of fruits and number of seeds per fruit, and substantial additional effort is required to collect and process data on two components compared to just one  [e.g. @Maddox1983; @Verhoeven2004; @Hall2006; @agren_reciprocal_2012; @leinonen2011local]. If investment in fruit production is negatively correlated with investment in seed production per fruit, relying on estimates of only one of these components of fecundity will overestimate variation in total fecundity. If investment in seed and fruit production are positively correlated, the opposite would be true.

Third, theory predicts a trade-off between investment in individual offspring and the total number of offspring [@Lack1954; @Smith1974]. In plants this would be expressed as a negative correlation between seed size and number [@Harper1970; @Leishman2000]. Selection for larger seeds may thus constrain the evolution of increased fecundity. Negative correlations between seed size and number have been documented across species [@Sera2004] and within crop species [@Sadras2007]. Meanwhile studies within natural plant populations have found positive, negative, and negligible correlations between seed size and number [@Silvertown1989; @Venable1992]. In Arabidopsis thaliana, trade-offs between seed size and number have been observed for recombinant lines grown in controlled environments [@alonso1999natural; @gnan2014genetic], and both linkage mapping and mutant screens have found pleiotropic or closely linked QTL with antagonistic effects on seed size and number [@alonso1999natural; @van2012comparative; @gnan2014genetic]. Available data suggest that a negative genetic correlation between seed size and number can constrain the evolution of fecundity.

Previous work on the genetics of correlations among fitness components and seed size in A. thaliana has largely been conducted in controlled conditions, and it is unclear whether results reflect the situation in natural environments [@alonso1999natural; @van2012comparative; @gnan2014genetic]. On the one hand, trade-offs may be more likely to be expressed in less benign natural environments, where resource availability is lower and stress in the form of frost, drought, and antagonistic biotic interaction more common. On the other hand, in genetically variable populations, variation in ability to meet the challenges of harsh environmental conditions may still result in sufficient variation in plant resource status to mask variation in allocation strategy. To understand properly the relationship between trade-offs, variation in condition and local adaptation, we need to compare genotypes in the environments from which they originate.

In this study, we investigate the contribution of individual components of fitness to estimates of local adaptation, and the genetic basis of correlations among components of fitness. We use a population of recombinant inbred lines (RILs) derived from a cross between two locally-adapted ecotypes of Arabidopsis thaliana from close to the southern (Italy) and northern (Sweden) margins of the native range in Europe. Reciprocal transplants have shown that the two source populations display strong adaptive differentiation expressed through higher survival and fruit production of the local ecotype [@agren_reciprocal_2012; @agren_genetic_2013], and there is some evidence that the local ecotype also produces more seeds per fruit compared to the non-local ecotype [@@agren_reciprocal_2012]. QTL mapping in the RIL population identified a total of 15 QTL for an estimate of overall fitness (number of fruits per seedling planted) at the sites of the two source populations [@agren_genetic_2013]. However, this estimate of overall fitness did not include possible variation in seed production per fruit, and it is therefore not clear how inclusion of this fitness component would affect estimates of selection against the non-local ecotype, correlations between fecundity and survival, or the genetic basis of fecundity and overall fitness.

Here, we quantify seed output per fruit and mean seed size per fruit of the parental ecotypes and of >300 RILs planted at the sites of the source populations in two years. We combine these data with previously published data on survival and fruit production to ask: (1) Does the local ecotype produce more seeds per fruit than does the non-local ecotype, which would result in an even larger estimate of selection against the non-local ecotype than an estimate previously reported based on survival and fruit production only? (2) Are there correlations between fecundity and survival, between components of fecundity (number of fruits and number of seeds per fruit), and between offspring number and size, and are these negative or positive? (3) Are there pleiotropic effects of QTL for number of seeds per fruit and seed mass on other components of fitness, and are these effects positive or negative?

# Materials and methods 

## Data collection 

We estimated seed traits for recombinant inbred lines (RIL) and parental accessions in reciprocal transplant experiments conducted at the native sites of the source populations in two years (2010-2011 and 2011-2012). These experiments have previously been described by @agren_reciprocal_2012 and @agren_genetic_2013, who quantified survival to reproduction, number of fruits per reproductive plant, and number of fruits per seedling planted for the parental ecotypes and 398 RIL. We expanded these data by quantifying the number of seeds per fruit (henceforth “seeds/fruit”) and mean seed mass per fruit. In each site × year combination we sampled a single mature fruit from between 1 and 12 plants per RIL and between 23 and 100 parental plants; sample sizes varied because lines varied in how well they survived in different sites and years. For each fruit, we counted the number of seeds and determined total seed mass to the nearest 0.01 mg on an AT261 balance (Mettler Toledo, Columbus, United States). We calculated mean seed mass as the mass of all seeds in a fruit, divided by the total number of seeds. We estimated genetic values for each line in each site × year combination as the mean across all individuals of the same RIL or parental ecotype. We could estimate genetic values for seeds/fruit and seed mass in 395 (Italy 2010), 398 (Italy 2011), 395 (Sweden 2010), and 394 (Sweden in 2011) of the total of 398 RILs. As such, there is unlikely to be substantial bias due to the RILs not included.

We combined data on seeds/fruit with previously published data to obtain estimates of overall fecundity and overall fitness that include information on seed number. In previous analyses of data from these experiments [@agren_reciprocal_2012; @agren_genetic_2013; @agren_flowering_time], fecundity was defined as number of fruits per reproductive plant (“fruits/RP”) and overall fitness as number of fruits per seedling planted (“fruits/seedling”). Here, we estimated the overall fecundity of reproductive plants (henceforth: “seeds/RP”) by multiplying individual fruits/RP by line-mean seeds/fruit. We chose to estimate fecundity this way because we did not have data on seeds/fruit for all individuals for which data on fruits/RP were available. Moreover, it was impractical to sample more than one fruit per plant, precluding any estimate of within-plant variation in seeds/fruit. We quantified total fitness as the number of seeds per seedling planted (“seeds/seedling”; zero for plants that did not survive to reproduce).

We estimated broad-sense heritability (H^2^) as the proportion of total phenotypic variation among individuals that is explained by RIL genotype in each site-year combination. We used a mixed-effect ANOVA estimated using the package *lme4* [@bates2015], with block as a fixed effect and RIL genotype as a random effect. To assess the uncertainty around these estimates we performed parametric bootstrapping on model parameters using the function *bootMer*, and estimated 95% confidence intervals as the 2.5% and 97.5% quantiles of 1000 bootstrap draws. We carried out data handling and statistical analyses in RStudio 1.1.442 using R `r paste(version$major, version$minor, sep=".")` [@RCT2015; @RStudioTeam2015]).

## Fitness differences between parental lines 

To assess the influence of different fitness components on estimates of adaptive differentiation, we quantified selection against the non-local ecotype by calculating selection coefficients based on individual components of fitness, and on estimates of overall fitness. We calculated selection coefficients $s=1-w_{min}/w_{max}$, where $w_{min}$ is the fitness of the less fit ecotype and $w_{max}$ that of the fitter ecotype. For cases where the non-local ecotype had higher fitness than the local ecotype, we multiplied the selection coefficient by -1. We calculated selection coefficients based on two estimates of overall fitness: fruits/seedling and seeds/seedling, reflecting fitness estimates excluding and including information on seeds/fruit. We also calculated selection coefficients based on survival, and on two components of fecundity (fruits/RP and seeds/fruit).

We estimated confidence intervals for selection coefficients by non-parametric bootstrapping. We drew 1000 bootstrap re-samples by sampling data with replacement from within experimental blocks. We calculated selection coefficients for each bootstrap sample and estimated 95% confidence intervals for each coefficient as the 2.5% and 97.5% quantiles of these values. We tested the null hypothesis that there is no adaptive differentiation using two-tailed p-values, calculated as twice the proportion of bootstrap values overlapping zero. It is more difficult to determine whether selection coefficients for the two measures of overall fitness, fruits/seedling and seeds/seedling, differ from one another because both estimates include common data on fruit number, and as such are not independent. Rather than perform a formal test, we simply asked whether the selection coefficient based on seeds/seedling was beyond the 95% confidence interval of that based on fruits/seedling. We compared differences between parental lines in mean seed mass in each site × year combination using Wilcox-signed-rank tests.

## Correlations between traits 

For each site × year combination we quantified the genetic correlations between pairs of traits by calculating Pearson correlation coefficients, r, between RIL means. A genetic correlation is the correlation between genetic values of genotypes, which are obtained by averaging over individuals within each genotype, whereas a phenotypic correlation is the correlation between phenotypes of individual plants, and is thus sensitive to environmental variation between individuals (Falconer & MacKay 1989). We estimated 95% confidence intervals around point estimates of genetic correlations by drawing 1000 non-parametric bootstrap samples from vectors of RIL means, recalculating correlation coefficients, and taking the 2.5% and 97.5% quantiles of the distribution of correlation coefficients across these resamples. We examined relationships between three pairs of traits: (1) survival and overall fecundity (seeds/RP), (2) the two components of fecundity (fruits/RP and seeds/fruit), and (3) overall fecundity and offspring size (mean seed mass).

## QTL mapping 

We mapped QTL for fitness and its components using the *R/qtl* package in R [@Broman2003; @Broman2009] using additional visualisation tools from the package *arghqtl* [@ellis_arghqtl]. We mapped QTL for seed mass, seeds/fruit, seeds/RP, and seeds/seedling. Mapping results for survival, fruits/RP and fruits/seedling were previously reported by @agren_genetic_2013 based on 398 RILs. However, the number and positions of QTL detected can be affected by the number of RILs included because different subsets of lines contain different recombination events. To allow comparisons of QTL positions and examination of evidence of pleiotropic QTL effects, we therefore repeated QTL mapping for survival, fruits/RP and fruits/seedling including only those RILs with information on seed size and number in each site × year combination as described in "Data collection".

We performed mapping based on RIL mean data for each site × year combination separately. We used Haley–Knott regression using genotype probabilities of the genetic markers and pseudomarkers in gaps >2 cM [@Haley1992]. We performed a two-QTL scan of the genome with 10,000 permutations of the phenotypic data to determine 5% LOD-significance thresholds for inclusion of QTL and epistatic interactions [@doerge1996permutation; @Broman2009]. Based on these thresholds, we used *R/qtl*’s automated stepwise model selection procedure to identify significant additive QTL and epistatic interactions. We applied a quantile normal transformation to phenotypes before model selection. Finally, we fitted a multiple-QTL model to untransformed data to calculate, for each locus, the proportion of the total phenotypic variance among RILs explained (PVE), and the effect size (in units of the trait) of a substitution of the Swedish homozygous genotype.

To investigate whether QTL showed pleiotropic effects on multiple traits, we examined whether QTL for different traits map (co-localise) to the same region. There are currently no clear guidelines on how to formally delineate QTL in linkage-mapping studies, so we rely on a set of heuristic rules used in previous studies [@agren_genetic_2013; @dittmar2014flowering; @oakley2014qtl; @agren_flowering_time; @postma2018among]. We considered any pair of QTL to co-localise and represent the same QTL if the 95% Bayesian credible intervals for these estimates overlapped. Based on these criteria, we identified ‘pleiotropic’ regions associated with multiple traits if they contained co-localising QTL for two or more traits that were directly observed (fruits/RP, seeds/fruit, survival and seed mass). We excluded “poorly-defined” QTL whose credible intervals for QTL position were greater than one quarter of the length of the shortest chromosome (15.2cM) from assessments of co-localisation, because such QTL provide little information about position.

# Results 
## Number of seeds per fruit influences estimates of selection

```{r selection-correlations, fig.height=12/2.54, fig.width=11.2/2.54, fig.cap="Selection against the non-local ecotype and genetic correlations among components of fitness, and between mean seed mass and fecundity . Selection against the non-local ecotype was estimated based on three components of fitness: survival to reproduction, number of fruits per reproductive plant (Fruits/RP), and number of seeds per fruit (Seeds/fr), and based on two estimates of total fitness: number of fruits per seedling planted (Fruits/sdl), and number of seeds per seedling planted (Seeds/sdl) at the Italian and Swedish sites (A, B). Genetic correlations between components of fitness and between mean seed mass (SdMass) and  number of seeds per reproductive plant (Seeds/RP) were quantified as the correlations among RIL means at the Italian and Swedish sites (C,D). Error bars show 95% bootstrap confidence intervals. Positive selection coefficients indicate selection favouring the local ecotype, negative values selection favouring the non-local ecotype.", include=T}
# setEPS()
# postscript(file = 'figures/01_selection.eps', width=12/2.54, height = 9/2.54)
# pdf(file = 'figures/01_selection.pdf', width=16.9/2.54, height = 11/2.54)

par(mfrow=c(2,2), mar=c(4,4,1,0)+0.1, oma=c(1,0,1,1), cex=0.7)

##############`~
# SELECTION ####
##############`~
os <- 0.1
ix <- match(c("survival", "fruit_per_plant",  "number_per_fruit", "fruit_per_seed", "total_fitness"), parents$variable)
# Selection in Italy
plot(c(0.5,5.5), c(-0.4,1), type='n', axes=F,
     xlab = "",
     ylab='Selection coefficient',
     main="Italy")
axis(2, cex.axis=1, las=1)
axis(1, 1:5, c("Survival","Fruits/RP","Seeds/fr", "Fruits/sdl","Seeds/sdl"), las=2)
box(); grid()
abline(0,0, lty=2)
# Points for 2010
segments(1:length(ix)-os, sel_cis$it2010[1,], 1:length(ix)-os, sel_cis$it2010[2,])
points(1:length(ix)-os, parents$s[ix  ], pch=16, bg='white')
# Points for 2011
segments(1:length(ix)+os, sel_cis$it2011[1,], 1:length(ix)+os, sel_cis$it2011[2,])
points(1:length(ix)+os, parents$s[ix+1], pch=21, bg='white')
text(0.5, 1, "A", adj=c(0,1))

# Selection in Sweden
plot(c(0.5,5.5), c(-0.4,1), type='n', axes=F, xlab = "", ylab='', main="Sweden")
axis(2, cex.axis=1, las=1)
axis(1, 1:length(ix), c(c("Survival","Fruits/RP","Seeds/fr", "Fruits/sdl","Seeds/sdl")), las=2, cex.axis=0.9)
box(); grid()
abline(0,0, lty=2)
# Points for 2011
segments(1:length(ix)-os, sel_cis$sw2010[1,], 1:length(ix)-os, sel_cis$sw2010[2,])
points(1:length(ix)-os, parents$s[ix+2], pch=16, bg='white')
# Points for 2011
segments(1:length(ix)+os, sel_cis$sw2011[1,], 1:length(ix)+os, sel_cis$sw2011[2,])
points(1:length(ix)+os, parents$s[ix+3], pch=21, bg='white')
# legend('topright', c("2010", "2011"), pch=c(16, 21), bg='white')
text(0.5, 1, "B", adj=c(0,1))

# dev.off()


##################~
# CORRELATIONS ####
##################`
os <- 0.05
xv <- sort(c((1:3) - os, (1:3) + os))

# Italy 
plot(c(0.5, 3.5), c(-0.4, 0.6), type='n', axes=F,
     xlab = "",
     ylab='Correlation coefficient',
     main="")
axis(2, cex.axis=1, las=1)
axis(1, 1:3, c("Survival vs.\nSeeds/RP", "Seeds/fr vs.\nFruits/RP", "SdMass vs.\nSeeds/RP"), las=2, cex.axis=0.9)
box()
grid()
abline(0,0, lty=2)
ix <- grep("it", correlations$expt)
segments(xv, correlations$lower[ix], xv, correlations$upper[ix])
points(xv, correlations$obs[ix], pch=c(16,21), bg='white')
legend('bottomleft', c("2010", "2011"), pch=c(16, 21), bg='white')
# legend('topleft', "C", bty='n')
text(0.5, 0.6, "C", adj=c(0,1))

# Sweden
plot(c(0.5, 3.5), c(-0.4, 0.6), type='n', axes=F,
     xlab = "",
     ylab='',
     main="")
axis(2, cex.axis=1, las=1)
axis(1, 1:3, c("Survival vs.\nSeeds/RP", "Seeds/fr vs.\nFruits/RP", "SdMass vs.\nSeeds/RP"), las=2, cex.axis=0.9)
box()
grid()
abline(0,0, lty=2)
ix <- grep("sw", correlations$expt)
segments(xv, correlations$lower[ix], xv, correlations$upper[ix])
points(xv, correlations$obs[ix], pch=c(16,21), bg='white')
text(0.5, 0.6, "D", adj=c(0,1))

```

Selection through seeds/fruit favoured the local genotype in both years in Italy, and in one year in Sweden (figure \@ref(fig:selection-correlations)A and \@ref(fig:selection-correlations)B).
In Italy, the local ecotype produced significantly more seeds/fruit in both 2010
(means:
`r format(round(parents$it_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[1]`
vs
`r format(round(parents$sw_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[1]`)
and 2011 (means:
`r format(round(parents$it_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[2]`
vs
`r format(round(parents$sw_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[2]`).
In Sweden, the local ecotype produced significantly more seeds per fruit than the non-local ecotype in 2010 (means:
`r format(round(parents$sw_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[3]`
vs
`r format(round(parents$it_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[3]`)
but not in 2011 (means: 
`r format(round(parents$sw_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[4]`
vs
`r format(round(parents$it_parent[parents$variable == "seeds_per_fruit"],1), nsmall=1)[4]`).
Selection through seed number per fruit in Sweden in 2010 is a novel source of selection as none was previously detected for survival or fruits/RP [figure \@ref(fig:selection-correlations)B; @agren_reciprocal_2012]. The only significant source of selection in 2011 in Sweden was via survival [figure \@ref(fig:selection-correlations)B; @agren_genetic_2013]. 

The local ecotype had significantly higher overall fitness (expressed as number of seeds per seedling planted) compared to the non-local ecotype at both sites and in both years ($p \leq 0.001$ in all cases; figure \@ref(fig:selection)).
This advantage was more than twice as strong in Italy than in Sweden
(selection coefficient against the foreign ecotype, *s*, of
`r round(parents$s[parents$variable == "total_fitness"], 2)[1]`
vs. 
`r round(parents$s[parents$variable == "total_fitness"], 2)[3]`
in 2010;
*s*=
`r round(parents$s[parents$variable == "total_fitness"], 2)[2]`
vs.
`r round(parents$s[parents$variable == "total_fitness"], 2)[4]` 
in 2011 in Italy and Sweden respectively).

At the Italian site, the local ecotype consistently outperformed the non-local ecotype for all components of fitness (figure \@ref(fig:selection)).
The Italian ecotype had significantly higher survival, corresponding to selection coefficients of
`r round(parents$s[parents$variable == "survival"], 2)[1]` in 2010 and 
`r round(parents$s[parents$variable == "survival"], 2)[2]` in 2011
($p < 0.001$),
produced more fruits per reproductive plant
(*s* = 
`r round(parents$s[parents$variable == "fruit_per_plant"], 2)[1]` in 2010 and
`r round(parents$s[parents$variable == "fruit_per_plant"], 2)[2]` in 2011;
$p < 0.001$),
and produced more seeds per fruit
(*s* = 
`r round(parents$s[parents$variable == "number_per_fruit"], 2)[1]` in 2010 and 
`r round(parents$s[parents$variable == "number_per_fruit"], 2)[2]` in 2011;
$p < 0.001$)
than did the non-local Swedish ecotype.
When information on number of seeds per fruit was included in estimates of overall fitness, estimates of selection favouring the local ecotype in Italy increased by
$\Delta s=$
`r round(parents$s[parents$variable == "total_fitness" & parents$site == "italy" & parents$year ==2010] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "italy" & parents$year ==2010],  3)`
(`r round(((parents$s[parents$variable == "total_fitness" & parents$site == "italy" & parents$year ==2010] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "italy" & parents$year ==2010]) / parents$s[parents$variable == "fruit_per_seed" & parents$site == "italy" & parents$year ==2010]) * 100, 1)`%)
in 2010 and by $\Delta s=$
`r round(parents$s[parents$variable == "total_fitness" & parents$site == "italy" & parents$year ==2011] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "italy" & parents$year ==2011],  3)`
(`r round(((parents$s[parents$variable == "total_fitness" & parents$site == "italy" & parents$year ==2011] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "italy" & parents$year ==2011]) / parents$s[parents$variable == "fruit_per_seed" & parents$site == "italy" & parents$year ==2011]) * 100, 1)`%)
in 2011 compared to when fecundity was based on fruit production only (figure \@ref(fig:selection)).
In both cases, the selection coefficient based on number of seeds per seedling planted was beyond the 95% confidence interval of that based on number of fruits per seedling planted (figure \@ref(fig:selection))

At the Swedish site, the contributions of survival and the two components of fecundity to the overall advantage of the local ecotype varied among years.
In 2010, the local ecotype produced more seeds per fruit
(*s* = `r round(parents$s[parents$variable == "number_per_fruit"], 2)[3]`;
$p < 0.001$),
but no significant differences between the two ecotypes were detected in survival
(*s* = `r round(parents$s[parents$variable == "survival"], 2)[3]`;
$p=$ `r 2*round(mean(sel_boots$sw2010[,1] < 0), 3)`),
or number of fruits per reproductive plant
(*s* = `r round(parents$s[parents$variable == "fruit_per_plant"], 2)[3]`;
$p=$ `r 2* round(mean(sel_boots$sw2010[,2] > 0), 3)`).
In 2011 the local ecotype had higher survival
(*s* = `r round(parents$s[parents$variable == "survival"], 2)[4]`;
$p < 0.001$),
whereas neither number of fruits per reproductive plant
(*s* = `r round(parents$s[parents$variable == "fruit_per_plant"], 2)[4]`;
$p=$ `r 2* round(mean(sel_boots$sw2011[,2] > 0), 3)`)
nor number of seeds per fruit
(*s* = `r round(parents$s[parents$variable == "number_per_fruit"], 2)[4]`;
$p=$ `r 2* round(mean(sel_boots$sw2011[,3] > 0), 3)`)
differed between the two ecotypes.
Including information on number of seeds per fruit substantially increased the estimated selection against the non-local ecotype in Sweden in 2010 
 ($\Delta s=$
`r round(parents$s[parents$variable == "total_fitness" & parents$site == "sweden" & parents$year ==2010] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "sweden" & parents$year ==2010],  3)`;
`r round(((parents$s[parents$variable == "total_fitness" & parents$site == "sweden" & parents$year ==2010] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "sweden" & parents$year ==2010]) / parents$s[parents$variable == "fruit_per_seed" & parents$site == "sweden" & parents$year ==2010]) * 100,  0)`%),
but had a negligible influence in 2011
 ($\Delta s=$
`r round(parents$s[parents$variable == "total_fitness" & parents$site == "sweden" & parents$year ==2011] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "sweden" & parents$year ==2011],  3)`;
`r round(((parents$s[parents$variable == "total_fitness" & parents$site == "sweden" & parents$year ==2011] - parents$s[parents$variable == "fruit_per_seed" & parents$site == "sweden" & parents$year ==2011]) / parents$s[parents$variable == "fruit_per_seed" & parents$site == "sweden" & parents$year ==2011]) * 100, 1)`%;
figure \@ref(fig:selection)).

## QTL for seeds/fruit contribute to differences in fecundity

```{r qtl, fig.width=16.9/2.54, fig.height=16/2.54, fig.cap='QTL for fitness components, seed mass and estimates fo fitness. Lanes show QTL for number of fruits per reproductive plant (Fruits/RP), number of seeds per fruit (Seeds/fr), number of seeds per reproductive plant (Seeds/RP), seed mass (Sd mass), survival, fruits/seedling (Fruits/sdl) and seeds/seedling (seeds/sdl). Arrows indicate most-likely QTL position and direction of effect of the Swedish genotype (upward: increased phenotype; downward: decreased phenotype) in Italy (red, upper panels) and Sweden (blue, lower panels) in the 2010 (10) and 2011 (11) experiments respectively. Vertical bars show the 95% Bayesian credible intervals for QTL position. Open arrows show QTL with credible intervals wider than 15.2cM. Grey boxes indicate regions Q1 to Q11 harbouring QTL with pleiotropic effects on two or more of those traits that were directly observed (number of fruits per plant, seed number per fruit, survival and seed mass). Tracks to the left of chromosomes show marker positions in cM. "+" and "-" symbols indicate QTL detected for seeds/seedling not detected for fruits/seedling in the same year, and vice versa.'}

# pdf(file='figures/QTL_all_traits.pdf', width=16.9/2.54, height=16/2.54)

par(mfrow=c(2,1), mar=c(0,1,0,0)+0.1, oma=c(0,1,0,0), cex=0.8)

#~~~~~~~~~~~~~~~~~~~~~
# set up plot object #
#~~~~~~~~~~~~~~~~~~~~~
pos_plot <- plotQTL(1:5, mloc, nlanes = 14, left_gap = 2, right_gap = 2)
colvec <- rep(c("red3", "blue4"), each=10)
# Details of pleiotropy boxes
# boxes <- qtl_clusters$boxes[- c(2,6,7,8,10,11,12),]
boxes <- qtl_clusters$boxes[qtl_clusters$summary$n.qtl > 1,][-6,]
ix    <- match(boxes$chr, colnames(pos_plot$lane_margins))
pos   <- (boxes$lower + boxes$upper) /2

ts <- 0.2


#~~~~~~~~
# ITALY # 
#~~~~~~~~
plotQTL_base(pos_plot, by=20, maxy = 20, track_labels = paste("Chr.", 1:5))
text(-6.5, -40, "Marker distance (cM)", srt=90, xpd=NA, cex=0.9)
text(-2,17, "Italy", xpd=NA, adj=c(1, 1), cex=1.1, col="red3")
# indicate extent of overlapping QTL with grey boxes.
plotQTL_boxes(
  pos_plot,
  boxes,
  border = 'gray75',
  col = 'gray75',
  lwd = 2
)

pos_counter <- 1
for(t in c("frut", "seed", "tofu", "mass", "surv", "ffit", "tfit")){
  # plot QTL positions
  for(i in c("it2010", "it2011")){
    plotQTL_qtl(pos_plot, qtl_stepwise[[t]][[i]], qtl_fits[[t]][[i]],
                fill_cutoff = 15.2, lane = pos_counter, col='red3', cex=0.6)
    pos_counter <- pos_counter + 1
  }
}
# Lane dividers
# segments(pos_plot$lane_margins, 0,
#          pos_plot$lane_margins, rep(-pos_plot$track_lengths, each=11),
#          lty=2, col='gray50')
segments(pos_plot$lane_margins[seq(1,15,2),], 
         0,
         pos_plot$lane_margins[seq(1,15,2),],
         rep(-pos_plot$track_lengths, each=8),
         lty=1, col=1, lwd=0.5)
# segments(pos_plot$lane_margins[c(1,11),], 0,
#          pos_plot$lane_margins[c(1,11),], rep(-pos_plot$track_lengths, each=2),
#          lty=1, col=1, lwd=2)
# Lane labels
text(pos_plot$lane_margins[seq(2,15,2),], 15,
     c("Fruits/RP", "Seeds/fr", "Seeds/RP", "Sd mass", "Survival", "Fruits/sdl", "Seeds/sdl"),
     col=1, cex=0.8, srt=90, adj=c(0.5,0.5))
text(pos_plot$lane_centres, 1, c("10", "11"), cex=0.5, adj=c(0,0.5), srt=90, col='gray30')
# Label pleiotropy boxes
text(pos_plot$lane_margins[1,ix]- 0.7,
     -pos,
     paste("Q",1:nrow(boxes), sep=""),
     adj=c(0.5,0),
     cex=0.7,
     srt=90)
plotQTL_ladder(pos_plot)

# Labels for QTL found for seeds/sdl, but not fruits/sdl
os <- 1
text(pos_plot$lane_margins[15,3] + os, -19.705, "+")
# text(pos_plot$lane_margins[15,4] + os, -20.7, "+")
text(pos_plot$lane_margins[15,2] + os, -31.7, "+")
text(pos_plot$lane_margins[15,5] + os, -18.5, "+")
# Labels for QTL found for fruits/sdl, but not seeds/sdl
# text(pos_plot$lane_margins[15,4] + os, -50.0, "-")
text(pos_plot$lane_margins[15,3] + os, -1.7, "-")



#~~~~~~~~~
# SWEDEN #
#~~~~~~~~~
plotQTL_base(pos_plot, by=20, maxy = 20, track_labels = paste("Chr.", 1:5))
text(-6.5, -40, "Marker distance (cM)", srt=90, xpd=NA, cex=0.9)
text(-2,17, "Sweden", xpd=NA, adj=c(1, 1), cex=1.1, col="blue4")
plotQTL_boxes(
  pos_plot,
  boxes,
  border = 'gray75',
  col = 'gray75',
  lwd = 2
)
pos_counter <- 1
for(t in c("frut", "seed", "tofu", "mass", "surv", "ffit", "tfit")){
  # plot QTL positions
  for(i in c("sw2010", "sw2011")){
    plotQTL_qtl(pos_plot, qtl_stepwise[[t]][[i]], qtl_fits[[t]][[i]],
                fill_cutoff = 15.2, lane = pos_counter, col='blue4', cex=0.6)
    pos_counter <- pos_counter + 1
  }
}
# Lanes dividers
# segments(pos_plot$lane_margins, 0,
#          pos_plot$lane_margins, rep(-pos_plot$track_lengths, each=11),
#          lty=2, col='gray50')
segments(pos_plot$lane_margins[seq(1,15,2),], 0,
         pos_plot$lane_margins[seq(1,15,2),], rep(-pos_plot$track_lengths, each=8),
         lty=1, col=1, lwd=0.5)
plotQTL_ladder(pos_plot)
# Lane labels
text(pos_plot$lane_margins[seq(2,15,2),], 15,
     c("Fruits/RP", "Seeds/fr", "Seeds/RP", "Sd mass", "Survival", "Fruits/sdl", "Seeds/sdl"),
     col=1, cex=0.8, srt=90, adj=c(0.5,0.5))
text(pos_plot$lane_centres, 1, c("10", "11"), cex=0.5, adj=c(0,0.5), srt=90, col='gray30')
# Label pleiotropy boxes
text(pos_plot$lane_margins[1,ix]- 0.7,
     -pos,
     paste("Q",1:nrow(boxes), sep=""),
     adj=c(0.5,0),
     cex=0.7,
     srt=90)

# Labels for QTL found for seeds/sdl, but not fruits/sdl
os <- 1
text(pos_plot$lane_margins[15,1] + os, -13.483, "+")
text(pos_plot$lane_margins[15,1] + os, -22.704, "+")
text(pos_plot$lane_margins[15,1] + os, -75.4, "+")
text(pos_plot$lane_margins[15,4] + os, -41.050, "+")
# Labels for QTL found for fruits/sdl, but not seeds/sdl
text(pos_plot$lane_margins[15,3] + os, -10.7, "-")
text(pos_plot$lane_margins[15,2] + os, -58.6, "-")
text(pos_plot$lane_margins[15,5] + os, -27.1, "-")
text(pos_plot$lane_margins[15,5] + os, -60.6, "-")

# dev.off()
```


We found QTL for seeds/fruit across all five chromosomes that together explained between 24.3% and 29.0% of the variance in mean seeds/fruit among RILs in each site-year combination (figure \@ref(fig:qtl), tables \@ref(tab:individual-seed-qtl) and \@ref(tab:epistatic-qtl-table)). For all QTL detected in Italy, the non-local Swedish allele was associated with fewer seeds per fruit. In Sweden, the local allele was associated with an increase in seeds/fruit in all four QTL detected in 2010, and in three out of five QTL in 2011. For the other two QTL the local allele was associated with a decrease in seeds/fruit (figure \@ref(fig:qtl)). Swedish alleles at QTL for seeds/fruit were thus associated with reduced seed output per fruit in Italy, whereas the direction of effects varied in Sweden. In addition, we detected two pairs of loci showing significant epistatic interactions for seeds/fruit in Sweden (figure \@ref(fig:epistatic-seed), table \@ref(tab:epistatic-qtl-table)). In 2010, RILs with at least one Swedish genotype at 1@22.7 or 5@70.6 produced more seeds per fruit compared to plants with the Italian genotype at both loci. In 2011, RILs with Swedish genotypes at both 1@61.1 and 3@21.0 showed the lowest fitness of any genotype combinations, and the highest fitness was recorded for RILs that were Swedish at 1@61.1 and Italian at 3@21.0.

Including information on seed number per fruit in estimates of overall fitness affected the number of fitness QTL detected (figure \@ref(fig:qtl), tables \@ref(tab:individual-ffit-qtl and @\ref(tab:individual-tfit-qtl))). Most QTL for seeds/seedling corresponded to a nearby QTL for fruits/seedling, and are likely to reflect the same QTL. However, in Italy, when fitness is quantified as seeds/seedling we detected four additional QTL that were not detected in the same year when fitness is defined as fruits/seedling (indicated with “+” in figure 2), but did not detect one QTL that was previously found for fruits/seedling (indicated with “-” in figure 2). In Sweden, none of the four QTL detected for number seeds per seedling planted in 2010 were detected when fitness is quantified as fruits/seedling. Four QTL detected for fruits/seedling were not detected for seeds/seedling. When information on seed number is incorporated into estimates of overall fitness, there is thus a net gain in the number of fitness QTL detected, although this is offset by the loss of some QTL detected when fitness is estimated based on survival and fruit production only.

## Limited differentiation in seed mass

Differences in seed mass between the two ecotypes depended on both site and year (supporting figure 1).
In 2010, The Italian parent produced larger seeds than did the Swedish parent at both the Italian site
(means $\pm$ SE, Italian ecotype,
`r round(parents$it_parent[parents$variable == "seed_mass" & parents$site == "italy" & parents$year == "2010"], 2)`
$\pm$
`r round(parents$it_se[    parents$variable == "seed_mass" & parents$site == "italy" & parents$year == "2010"], 2)`
µg; Swedish ecotype,
`r round(parents$sw_parent[parents$variable == "seed_mass" & parents$site == "italy" & parents$year == "2010"], 2)`
$\pm$
`r round(parents$sw_se[    parents$variable == "seed_mass" & parents$site == "italy" & parents$year == "2010"], 2)`
µg; 
$W =$ `r mass_wilcox$italy_2010$statistic`,
$p=$ `r round(mass_wilcox$italy_2010$p.value, 3)`) 
and Swedish site
(Italian ecotype,
`r round(parents$it_parent[parents$variable == "seed_mass" & parents$site == "sweden" & parents$year == "2010"], 2)`
$\pm$
`r format(round(parents$it_se[    parents$variable == "seed_mass" & parents$site == "sweden" & parents$year == "2010"], 2), nsmall=2)`
µg; Swedish ecotype,
`r round(parents$sw_parent[parents$variable == "seed_mass" & parents$site == "sweden" & parents$year == "2010"], 2)`
$\pm$
`r round(parents$sw_se[    parents$variable == "seed_mass" & parents$site == "sweden" & parents$year == "2010"], 2)`
µg;
$W =$ `r mass_wilcox$sweden_2010$statistic`,
$p=$ `r round(mass_wilcox$sweden_2010$p.value, 3)`).
No significant difference in seed mass between the two parental ecotypes was recorded in the second year at either the Italian site
(
$W =$ `r mass_wilcox$italy_2011$statistic`, $p=$ `r round(mass_wilcox$italy_2011$p.value, 3)`;
Swedish site:
$W =$ `r mass_wilcox$sweden_2011$statistic`, $p=$ `r round(mass_wilcox$sweden_2011$p.value, 3)`;
supporting figure 1).
Both ecotypes produced larger seeds at the site in Sweden compared to that in Italy.

## Positive correlations dominate among fitness components

```{r correlation-no-outliers}
cortests2 <- list(
  cor.test(ril_tofu$it2010[ril_mass$it2010 < 35], ril_mass$it2010[ril_mass$it2010 < 35], method = 's'),
  cor.test(ril_tofu$it2011[ril_mass$it2011 < 35], ril_mass$it2011[ril_mass$it2011 < 35], method = 's'),
  cor.test(ril_tofu$sw2010[ril_mass$sw2010 < 35], ril_mass$sw2010[ril_mass$sw2010 < 35], method = 's'),
  cor.test(ril_tofu$sw2011[ril_mass$sw2011 < 35], ril_mass$sw2011[ril_mass$sw2011 < 35], method = 's')
  )

corsumm2 <- data.frame(experiment = siteyear,
                       rho = round(sapply(cortests2, function(x) x$estimate),2),
                       p   = round(sapply(cortests2, function(x) x$p.value), 3))
```

We found positive correlations between number of fruits per reproductive plant and number of seeds per fruit, as well as between survival and number of seeds per reproductive plant in both years in Italy, and in Sweden in 2011 ($r_P \geq$
`r round(min(correlations$obs[correlations$traits %in% c("frut_vs_numb", "tofu_vs_surv") & correlations$expt != "sw2010"]), 2)`;
$p < 0.0001$;
figure \@ref(fig:correlations), supporting figure 2).
In Sweden in 2010, the positive correlation between number of fruits per reproductive plant and number of seeds per fruit was weaker but still significant (
$r_P=$ `r round(correlations$obs[correlations$trait == "frut_vs_numb" & correlations$expt == "sw2010"], 2)`,
$p<0.0001$),
while survival and number of seeds per reproductive plant were not significantly correlated
($r_P=$ `r round(correlations$obs[correlations$trait == "tofu_vs_surv" & correlations$expt == "sw2010"], 2)`,
$p=$ `r round(correlations$obs[correlations$trait == "tofu_vs_surv" & correlations$expt == "sw2010"], 3)`). 

In Sweden in 2010, seed mass was negatively correlated with fecundity
($r_P=$ `r format(round(correlations$obs[correlations$trait == "tofu_vs_mass" & correlations$expt == "sw2010"], 2), nsmall=2)`,
$p \leq 0.0001$),
whereas no significant correlation was detected between seed mass and fecundity in Sweden in 2011, nor in Italy in either year
($r_P \leq$ `r round(max(correlations$obs[correlations$trait == "tofu_vs_mass" & correlations$expt != "sw2010"]),2)`,
$p \geq$ `r round(min(correlations$p[correlations$trait == "tofu_vs_mass" & correlations$expt != "sw2010"]),3)`;
figure \@ref(fig:correlations), supporting figure 2).
A small number of lines showed unusually high seed mass and low fecundity (supporting figure 2).
To examine whether these lines unduly inflate estimates of correlations between seed mass and fecundity, we repeated the analyses excluding lines with mean seed mass greater than 35µg.
Neither test statistics nor p-values changed substantially when these lines were excluded
(Sweden 2010: $r_P=$ `r round(cortests2[[3]]$estimate, 2)`, $p=$ `r round(cortests2[[3]]$p.value, 3)`;
Sweden 2011: $r_P=$ `r round(cortests2[[4]]$estimate, 2)`, $p=$ `r round(cortests2[[4]]$p.value, 3)`;
Italy 2010: $r_P=$ `r round(cortests2[[1]]$estimate, 2)`, $p=$ `r round(cortests2[[1]]$p.value, 3)`;
Italy 2011: $r_P=$ `r round(cortests2[[2]]$estimate, 2)`, $p=$ `r round(cortests2[[2]]$p.value, 3)`).

## QTL for seed number per fruit contribute to differences in fecundity

```{r qtl-pleiotropy, fig.width=16.9/2.54, fig.height=16/2.54, fig.cap="QTL for fecundity, seed mass, and survival. Lanes show QTL for number of fruits per reproductive plant (Fruits/RP), number of seeds per fruit (Seeds/fr), number of seeds per reproductive plant (Seeds/RP), seed mass (Sd mass) and survival. Arrows indicate most-likely QTL position and the effect of the Swedish genotype (upward: increased phenotype; downward: decreased phenotype) in Italy (red, upper panels) and Sweden (blue, lower panels) in the 2010 (10) and 2011 (11) experiments respectively. Vertical bars show the 95\\% Bayesian credible intervals for QTL position. Open arrows show QTL with credible intervals wider than 15.2cM. Grey boxes indicate regions harbouring QTL with pleiotropic effects on two or more of those traits that were directly observed (number of fruits per plant, seed number per fruit, survival and seed mass) (Q1-Q11). Tracks to the left of chromosomes show marker positions."}
# setEPS()
# postscript('figures/03_QTL_pleiotropy.eps', width=16.9/2.56, height = 16/2.54)
# pdf('figures/QTL_pleiotropy_separate_sites.pdf', width=16.9/2.56)
par(mfrow=c(2,1), mar=c(0,1,0,0)+0.1, oma=c(0,1,0,0), cex=0.8)

#~~~~~~~~~~~~~~~~~~~~~
# set up plot object #
#~~~~~~~~~~~~~~~~~~~~~
pos_plot <- plotQTL(1:5, mloc, nlanes = 10, left_gap = 2, right_gap = 2)
colvec <- rep(c("red3", "blue4"), each=10)
# Details of pleiotropy boxes
# boxes <- qtl_clusters$boxes[- c(2,6,7,8,10,11,12),]
boxes <- qtl_clusters$boxes[qtl_clusters$summary$n.qtl > 1,][-6,]
ix    <- match(boxes$chr, colnames(pos_plot$lane_margins))
pos   <- (boxes$lower + boxes$upper) /2

ts <- 0.2


#~~~~~~~~
# ITALY # 
#~~~~~~~~
plotQTL_base(pos_plot, by=20, maxy = 20, track_labels = paste("Chr.", 1:5))
text(-6.5, -40, "Marker distance (cM)", srt=90, xpd=NA, cex=0.9)
text(-2,17, "Italy", xpd=NA, adj=c(1, 1), cex=1.1, col="red3")
# indicate extent of overlapping QTL with grey boxes.
plotQTL_boxes(
  pos_plot,
  boxes,
  border = 'gray75',
  col = 'gray75',
  lwd = 2
)

pos_counter <- 1
for(t in c("frut", "seed", "tofu", "mass", "surv")){
  # plot QTL positions
  for(i in c("it2010", "it2011")){
    plotQTL_qtl(pos_plot, qtl_stepwise[[t]][[i]], qtl_fits[[t]][[i]],
                fill_cutoff = 15.2, lane = pos_counter, col='red3')
    pos_counter <- pos_counter + 1
  }
}
# Lane dividers
# segments(pos_plot$lane_margins, 0,
#          pos_plot$lane_margins, rep(-pos_plot$track_lengths, each=11),
#          lty=2, col='gray50')
segments(pos_plot$lane_margins[seq(1,11,2),], 0,
         pos_plot$lane_margins[seq(1,11,2),], rep(-pos_plot$track_lengths, each=6),
         lty=1, col=1, lwd=0.5)
# segments(pos_plot$lane_margins[c(1,11),], 0,
#          pos_plot$lane_margins[c(1,11),], rep(-pos_plot$track_lengths, each=2),
#          lty=1, col=1, lwd=2)
# Lane labels
text(pos_plot$lane_margins[seq(2,11,2),], 15,
     c("Fruits/RP", "Seeds/fr", "Seeds/RP", "Sd mass", "Survival"), col=1, cex=0.8, srt=90, adj=c(0.5,0.5))
text(pos_plot$lane_centres, 1, c("10", "11"), cex=0.6, adj=c(0,0.5), srt=90, col='gray30')
# Label pleiotropy boxes
text(pos_plot$lane_margins[1,ix]- 0.7,
     -pos,
     paste("Q",1:nrow(boxes), sep=""),
     adj=c(0.5,0),
     cex=0.7,
     srt=90)
plotQTL_ladder(pos_plot)



#~~~~~~~~~
# SWEDEN #
#~~~~~~~~~
plotQTL_base(pos_plot, by=20, maxy = 20, track_labels = paste("Chr.", 1:5))
text(-6.5, -40, "Marker distance (cM)", srt=90, xpd=NA, cex=0.9)
text(-2,17, "Sweden", xpd=NA, adj=c(1, 1), cex=1.1, col="blue4")
plotQTL_boxes(
  pos_plot,
  boxes,
  border = 'gray75',
  col = 'gray75',
  lwd = 2
)
pos_counter <- 1
for(t in c("frut", "seed", "tofu", "mass", "surv")){
  # plot QTL positions
  for(i in c("sw2010", "sw2011")){
    plotQTL_qtl(pos_plot, qtl_stepwise[[t]][[i]], qtl_fits[[t]][[i]],
                fill_cutoff = 15.2, lane = pos_counter, col='blue4')
    pos_counter <- pos_counter + 1
  }
}
# Lanes dividers
# segments(pos_plot$lane_margins, 0,
#          pos_plot$lane_margins, rep(-pos_plot$track_lengths, each=11),
#          lty=2, col='gray50')
segments(pos_plot$lane_margins[seq(1,11,2),], 0,
         pos_plot$lane_margins[seq(1,11,2),], rep(-pos_plot$track_lengths, each=6),
         lty=1, col=1, lwd=0.5)
plotQTL_ladder(pos_plot)
# Lane labels
text(pos_plot$lane_margins[seq(2,11,2),], 15,
     c("Fruits/RP", "Seeds/fr", "Seeds/RP", "Sd mass", "Survival"), col=1, cex=0.8, srt=90, adj=c(0.5,0.5))
text(pos_plot$lane_centres, 1, c("10", "11"), cex=0.6, adj=c(0,0.5), srt=90, col='gray30')
# Label pleiotropy boxes
text(pos_plot$lane_margins[1,ix]- 0.7,
     -pos,
     paste("Q",1:nrow(boxes), sep=""),
     adj=c(0.5,0),
     cex=0.7,
     srt=90)

# dev.off()
```

Swedish alleles at QTL for number of seeds per fruit were associated with reduced seed output per fruit in Italy, whereas the direction of effects varied in Sweden.
In Italy, we identified a total of six distinct QTL for number of seeds per fruit, of which three were detected in both years (figure \@ref(fig:qtl-pleiotropy), supporting table 1).
For all QTL, the non-local Swedish allele was associated with fewer seeds per fruit.
In Sweden, we identified a total of six distinct QTL for number of seeds per fruit, of which three were detected in both years.
At four of these loci, the local Swedish allele was associated with an increase in number of seeds per fruit, whereas at the other two it was associated with fewer seeds per fruit.

QTL for overall fecundity (number of seeds per reproductive plant) could be explained by QTL for individual components of fecundity.
In Italy we identified a total of 12 distinct QTL for overall fecundity, of which four were detected in both years (figure \@ref(fig:qtl-pleiotropy); supporting table 2).
All of these loci colocalised with QTL for either number of fruits per reproductive plant or number of seeds per fruit, and seven loci colocalised with QTL for both components of fecundity (figure \@ref(fig:qtl-pleiotropy); supporting table 2-3). 
In Sweden, we detected a total of seven distinct QTL for number of seeds per reproductive plant, one of which was detected in both years.
All of these loci colocalised with QTL for either number of fruits per reproductive plant or number of seeds per fruit, and four colocalised with QTL for both components of fecundity (figure \@ref(fig:qtl-pleiotropy); supporting table 2-3).

Most QTL for number of seeds per seedling planted corresponded to a QTL detected for fruit number per seedling planted (figure \@ref(fig:fitness-qtl); supporting tables 4-5), but including information about seed output per fruit did result in some changes in the map positions of QTL for overall fitness.
In Sweden in 2010, when fitness differences between the parental genotypes were mainly expressed through differences in number of seeds per fruit, the three QTL detected for number of seeds per seedling planted were not observed when fitness was quantified as fruit number per seedling planted (figure \@ref(fig:fitness-qtl)).
Moreover, two QTL for overall fitness were detected in Italy in 2010 (chr. 3 18.0 cM; chr 5. 12.9) and one in Italy in 2011 (chr. 1 13.5 cM), which did not overlap with QTL for number of fruits per seedling planted, but colocalised with QTL for number of seeds per reproductive plant in the same year.
The position of one of the former QTL for overall fitness (chr 5. 12.0) was shifted by 4.5 cM compared to the closely located QTL for number of fruits per seedling planted (chr. 5, 7.5 cM; figure \@ref(fig:fitness-qtl)).
Other changes included the absence of QTL for number of seeds per seedling planted at the positions of four QTL detected for fruit number per seedling planted (Italy 2010, chr 3. 54.1 cM, chr 4. 50.0 cM; Italy 2011, chr 3. 2.2 cM; Sweden 2010, chr 3. 9.6 cM; figure \@ref(fig:fitness-qtl)). Including information about seed output per fruit in estimates of overall fitness thus allowed some additional fitness QTL to be detected, but also affected estimates of a few of the QTL observed when fitness was quantified based on survival and fruit production only.

```{r fitness-qtl, fig.width=8/2.56, fig.height = 15/2.54, fig.cap="QTL for number of fruits per seedling planted (Fruits/sdl) and number of seeds per seedling planted (Seeds/sdl) in each of the four site \U00D7 year combinations. Arrows indicate most-likely QTL position and the effect of the Swedish genotype (upward: increased phenotype; downward: decreased phenotype) in Italy (red) and Sweden (blue). Vertical bars show the 95% Bayesian credible intervals for QTL position. Open arrows show QTL with credible intervals wider than 15.2cM. Tracks to the left of chromosomes show marker positions."}
# setEPS()
# postscript('figures/04_QTL_survival_fitness.eps', width=16.9/2.56, height = 12/2.54)
# pdf('figures/05_QTL_fitness.pdf', width=16.9/2.56, height = 12/2.54)


colvec=c("red3", "red3","blue4","blue4")

par(mfrow=c(2,1), mar=c(0,3,0,0)+0.1, oma=c(0,0,0,0), cex=0.8)

pos_plot <- plotQTL(1:5, mloc, nlanes = 4, left_gap = 4, right_gap = 1)
plotQTL_base(pos_plot, by=20, maxy = 30, track_labels = paste("Chr.", 1:5))
text(-2,22, "Italy", xpd=NA, adj=c(1, 1), cex=1.1, col="red3")

# text(35, -95, "Chromosome")

# Plot QTL fitness boxes from Agren 2013.
# plotQTL_boxes(pos_plot, boxes[,c("chr", "lower", "upper")], col='gray80', border=F)
# Plot QTL in each lane
pos_counter <- 1
for(i in c("it2010", "it2011")){
  plotQTL_qtl(pos_plot, qtl_stepwise$ffit[[i]], qtl_fits$ffit[[i]],
              fill_cutoff = 15.2, lane = pos_counter, col=colvec[pos_counter])
  plotQTL_qtl(pos_plot, qtl_stepwise$tfit[[i]], qtl_fits$tfit[[i]],
              fill_cutoff = 15.2, lane = pos_counter+2, col=colvec[pos_counter])
  pos_counter <- pos_counter + 1
}
# Lanes dividers
segments(pos_plot$lane_margins[seq(1,5,2),], 0,
         pos_plot$lane_margins[seq(1,5,2),], rep(-pos_plot$track_lengths, each=3),
         lty=1, col=1, lwd=0.5)
# Lane labels
text(pos_plot$lane_centres, 2, c("10", "11"), cex=0.5, adj=c(0.5,0.5), srt=90, col='gray30')
# text(pos_plot$lane_margins[seq(2,5,2),], 6, c("It", "Sw"), cex=0.8, adj=c(0.5,0.5), col=c('red3','blue4'))
text(pos_plot$lane_margins[c(2,4),], 10, labels = c("Fruits/sdl", "Seeds/sdl"), cex=0.8, srt=90, adj = 0)

plotQTL_ladder(pos_plot, marker_tick_width = 1)



# SWEDEN
plotQTL_base(pos_plot, by=20, maxy = 30, track_labels = paste("Chr.", 1:5))
text(-2,22, "Sweden", xpd=NA, adj=c(1, 1), cex=1.1, col="blue4")
pos_counter <- 1
for(i in c("sw2010", "sw2011")){
  plotQTL_qtl(pos_plot, qtl_stepwise$ffit[[i]], qtl_fits$ffit[[i]],
              fill_cutoff = 15.2, lane = pos_counter, col="blue4")
  plotQTL_qtl(pos_plot, qtl_stepwise$tfit[[i]], qtl_fits$tfit[[i]],
              fill_cutoff = 15.2, lane = pos_counter+2, col="blue4")
  pos_counter <- pos_counter + 1
}
# Lanes dividers
segments(pos_plot$lane_margins[seq(1,5,2),], 0,
         pos_plot$lane_margins[seq(1,5,2),], rep(-pos_plot$track_lengths, each=3),
         lty=1, col=1, lwd=0.5)
plotQTL_ladder(pos_plot, marker_tick_width = 1)
# Lane labels
text(pos_plot$lane_centres, 2, c("10", "11"), cex=0.5, adj=c(0.5,0.5), srt=90, col='gray30')
# text(pos_plot$lane_margins[seq(2,5,2),], 6, c("It", "Sw"), cex=0.8, adj=c(0.5,0.5), col=c('red3','blue4'))
text(pos_plot$lane_margins[c(2,4),], 10, labels = c("Fruits/sdl", "Seeds/sdl"), cex=0.8, srt=90, adj = 0)
# dev.off()

```

## QTL for seed size vary in the direction of effects

In Italy, we identified two QTL for seed mass in 2010 and five in 2011 (figure \@ref(fig:qtl-pleiotropy), supporting table 6).
At two of the QTL detected in 2011, the Swedish allele was associated with an increase in seed mass, and at three with a decrease. In Sweden, we detected five QTL for seed mass in each year, one of which was detected in both years. The local Swedish alleles at three of the five QTL detected in 2010, and at one of the five QTL detected in 2011, were associated with a decrease in seed mass.

## QTL show positive pleiotropy

QTL for components of fitness that could be resolved to within 15.2 cM tended to map to one of eight distinct regions of the linkage map (Q1-Q8, indicated in grey in figure \@ref(fig:qtl-pleiotropy); supporting tables 1-3, 6-7).
Four regions in Italy (Q2, Q3, Q4, Q6, Q8) and one region in Sweden (Q8) included QTL for both survival and overall fecundity (number of seeds per reproductive plant).
Meanwhile, five regions in Italy (Q1, Q2, Q4, Q6, Q7) and two regions in Sweden (Q2, Q8) included QTL for both number of fruits per reproductive plant and seed number per fruit.
This indicates that loci in these regions have pleiotropic effects on multiple components of fitness, and/or that these regions harbour multiple loci in tight linkage that affect individual fitness components.

We next examined whether alleles at QTL within these regions were associated with positive or negative pleiotropy between survival and fecundity, and between components of fecundity.
In Italy, the non-local Swedish alleles at QTL in regions showing pleiotropic effects on survival and number of seeds per reproductive plant or on number of fruits per plant and number of seeds per fruit were associated with a decrease in both fitness components (figure \@ref(fig:qtl-pleiotropy)).
In Sweden, the local allele at the well-defined QTL for survival (Q8) was associated with increases in both survival and seed number per reproductive plant.
Local alleles at QTL detected for number of fruits per reproductive plant and number of seeds per fruit were associated with an increase in both fitness components (Q8), or with a decrease in both components (Q2).
Thus, when QTL for both survival and fecundity or both components of fecundity map to the same places, the Swedish allele was associated with effects in the same direction, even if the direction varied.
This indicates that these QTL have positive pleiotropic effects on components of fitness.

QTL for seed mass tended to map to regions harbouring QTL for fitness QTL.
Well-defined QTL for seed mass mapped to regions Q1, Q5 and Q6 in Italy, and to Q1, and Q6 in Sweden (figure \@ref(fig:qtl-pleiotropy)).
The only well-defined seed-mass QTL that did not colocalise with QTL for at least one other trait is the locus detected on chr. 4, 14.1 cM in Italy in 2011.
At Q1 and Q6 in Sweden we detected two distinct QTL for seed mass within the intervals of the regions identified as pleiotropic.

Pleiotropic interactions between fecundity and seed mass were more variable than among components of fitness.
In Italy, the Swedish allele at one QTL (Q5) was associated with a decrease in both seed mass and seed number per reproductive plant, whereas at Q6 the Swedish allele was associated with an increase in seed mass but a decrease in seed number per reproductive plant.
In Sweden, only the QTL on chr 1. at 21.9 cM colocalised with a QTL for number of seeds per reproductive plant, and at this locus the local allele was associated with a decrease in seed mass but an increase in seed number per reproductive plant.
QTL for seed mass thus showed both positive and negative pleiotropic associations with fecundity (figure \@ref(fig:qtl-pleiotropy)).

## Epistasis

We observed five instances of pairs of loci that showed significant epistatic interactions (supporting table 8, supporting figures 3-5).
Firstly, in Sweden in 2011 we found an interaction between QTL detected in region Q2 and within 2 cM of the marker at 23 cM on chr. 3 for number of seeds per fruit, number of seeds per reproductive plant and number of seeds per seedling planted that explained 8.0%, 9.3% and 7.3% of the overall phenotypic variances respectively.
This corresponds to the pair of epistatic loci detected for number of fruits per seedling planted in the previous analysis by @agren_genetic_2013.
In addition, we detected an epistatic interaction in Italy in 2011 between QTL for number of seeds per fruit in regions Q1 and Q6 explaining 2.7% of the variance, and another interaction in Italy in 2010 between QTL for number of seeds per seedling planted in region Q6 and the marker on chr. 3 at 18.0 cM explaining 0.58% of the phenotypic variance.

# Discussion

The present study demonstrates that genetic differences influencing number of seeds per fruit can make an important contribution to adaptive differentiation and the genetic basis of fitness variation among natural populations of *Arabidopsis thaliana*. In a reciprocal transplant between an Italian population located close to the southern margin of the European native range and a Swedish population located close to the northern range margin, the local ecotype produced more seeds per fruit than did the non-local ecotype in three of four site × year combinations (figure \@ref(fig:selection)). Including information about number of seeds per fruit thus increased the estimated magnitude of the fitness advantage of the local ecotype compared to estimates based on differences in fruit production and survival alone. Genetic correlations between fecundity and survival, and between components of fecundity (number of fruits per reproductive plant and number of seeds per fruit) were generally positive, whereas the correlation between fecundity and seed size was significant (and negative) in only one of four site × year combinations (figure \@ref(fig:correlations)). The genetic correlations were reflected in widespread pleiotropic effects of QTL for fecundity and survival, with allelic effects typically in the same directions (figure \@ref(fig:qtl-pleiotropy)). Below we discuss the results in relation to processes affecting adaptive differentiation and pleiotropic interactions among traits.

## Adaptive differentiation for seed number per fruit

The examination of variation in seed number per fruit provided several new insights into variation in fecundity and how this contributes to adaptive differentiation between these populations.
In both years in Italy, the local ecotype produced more seeds per fruit than did the non-local ecotype, and had a greater overall fitness advantage when fitness was estimated including information about number of seeds per fruit (figure \@ref(fig:selection)).
However, because selection through number of seeds per fruit was not as strong as selection through number of fruits per reproductive plant and survival (figure \@ref(fig:selection)), and number of seeds per fruit was strongly positively correlated with number of fruits per reproductive plant (figure \@ref(fig:correlations)), much of the adaptive differentiation expressed in Italy was captured by differences in number of fruits per seedling planted.
In Sweden on the other hand, number of seeds per fruit was the only fitness component for which the parental ecotypes differed in 2010 (figure \@ref(fig:selection)).
This was reflected in a significant overall advantage to the local ecotype that year that was only detected when this component was included in the estimate of overall fitness.
Despite differences between sites and between years, the results show that fecundity variation in *A. thaliana* is due to differences in both fruit production and number of seeds per fruit.

Including information on seed production into estimates of overall fitness allowed us to detect additional QTL for fitness. In particular, we detected three additional fitness QTL in Sweden in 2010, where only one was detected when fitness was based on survival and fruit production only. All three QTL colocalised with QTL for seed number per fruit (figures \@ref(fig:qtl-pleiotropy) and \@ref(fig:fitness-qtl)), and we found significant selection against the non-local Italian genotype through seed number per fruit but not through number of fruits per reproductive plant in Sweden in 2010 (figure \@ref(fig:selection)). This indicates that the effects of QTL on number of seeds per fruit were responsible for the differences in overall fitness associated with these loci.

Surprisingly, some QTL for fitness estimated as the number of fruits per seedling planted were not detected when fitness was estimated as number of seeds per seedling planted (figure \@ref(fig:fitness-qtl)).
One explanation for this could be that QTL have weakly negative pleiotropic effects on one or more combinations of number of seeds per fruit, number of fruits per reproductive plant and survival.
However, in almost all cases where pleiotropy was observed the effects were positive (figure \@ref(fig:qtl-pleiotropy)), so this seems unlikely.
Alternatively, our estimates of number of seeds per fruit might be less precise than for number of fruits per reproductive plant, which would inflate the residual variance of our estimates of number of seeds per seedling planted.
Although sample sizes for number of seeds per fruit and seed mass were much smaller than for survival and fruit production, this was not reflected in reduced heritability of seed traits (supporting figure 6), so this explanation also appears unlikely.
A third explanation is that there are many loci affecting fitness with effects close to the threshold of statistical significance, which would be consistent with classical population-genetic theory [@Fisher1930].
Such subtle effects would be sensitive to the precise way in which fitness is defined, as well as to fluctuations in environmental noise.
Because linkage mapping is designed to detect relatively few loci of large effect, this would cause some stochasticity in the loci detected and their map positions indicated by the stepwise-regression approach used in QTL mapping [@beavis1998qtl; @harrell2001regression; @Broman2009].
The apparent disappearance of fitness QTL when information on seed number is included could thus reflect a highly polygenic nature of QTL affecting fitness.

## Positive pleiotropic effects on multiple fitness components

Both genetic correlations in the RIL population and the QTL mapping showed evidence of positive pleiotropic effects on different components of fitness.
Correlations between components of fecundity, and between fecundity and survival were positive, except for Sweden in 2010 when no significant correlation was observed between fecundity and survival (figure \@ref(fig:correlations)).
Moreover, QTL for these components of fitness tended to map to the same regions of the genome, and allelic effects were in the same direction in all cases (figure \@ref(fig:qtl-pleiotropy)).
Taken together, the overall positive genetic correlations among phenotypes are reflected in positive pleiotropic effects of the underlying genetic loci.

The preponderance of positive genetic correlations and positive pleiotropy indicates that variation in overall vigour or "condition", where the fittest genotypes have more resources overall to invest across fitness components, predominates over variation in relative allocation to different components of fitness in the RIL population [@VanNoordwijk1986; @Houle1991]. This suggests that adaptive differentiation between the two focal populations largely reflects the fixation of variants that allowed for increased ability to acquire resources and grow under local environmental conditions, that in turn positively affect several components of fitness.
In a meta-analysis of local adaptation across 74 studies of plants, animals, fungi and protists, @Hereford2009 was not able to test explicitly for correlations between selection through components of fitness, but did demonstrate that the advantage of local populations was greater when estimated based on overall fitness than when estimated based on survival or fecundity only.
Although not a formal test, this pattern would be expected if selection acts to increase overall condition, causing components of fitness to be positively correlated within populations.
These observations indicate that adaptation frequently entails the evolution of increased condition in the local environment, and that while local adaptation is reflected as trade-offs in performance across environments, it may often also be associated with positive genetic correlations among fitness components within a given environment in situations where there is genetic variation for overall fitness.. 

Several traits may contribute to variation in resource status in *A. thaliana*. The Swedish ecotype has higher freezing tolerance [@oakley2014qtl] and a greater ability to optimize photosynthesis at cold, but non-freezing temperatures [@cohu2013minor; @adams2014associations; @oakley2017genetic]. This should provide a fitness advantage at the Swedish site where plants are exposed to cold conditions for an extended period, but may be associated with a fitness cost in a milder climate [@oakley2014qtl]. QTL affecting these traits can thus be expected to have pleiotropic effects on overall fitness and its components. QTL for phenological variation may also play a key role in variation in resource status. Previous work on the same populations has demonstrated strong selection to tune the timing of germination at both sites and flowering time in Italy to match the local climate [@akiyama2014conflicting; @postma_early_2016; @agren_flowering_time]. Differences in timing of life-history transitions should contribute to differences in resource status by allowing locally adapted ecotypes to grow and reproduce at the times best suited to local conditions. For example, @akiyama2014conflicting experimentally demonstrated that early germination allowed for faster autumn growth and increased winter survival at the Swedish site. QTL affecting both physiological and phenological traits can thus be expected to influence resource status and thereby have pleiotropic effects on several components of fitness.

## Pleiotropy and linkage

Two caveats need to be borne in mind regarding our results on the pleiotropic effects of QTL.
Firstly, resolution in a mapping population derived from a cross is limited by the number of recombination breakpoints, resulting in linkage disequilibrium between nearby markers.
As such, it is very difficult to distinguish a single pleiotropic locus from two or more tightly-linked loci affecting individual traits separately [@jiang1995multiple].
However, two loci with little recombination will tend to be co-inherited, and hence will segregate much like a single locus [@paaby2013many].
Therefore, from an evolutionary perspective, it makes little difference whether a QTL reflects a single pleiotropic locus or multiple strongly linked loci, because both traits will co-evolve in a similar manner in both cases.

A second caveat is that even if a QTL genuinely is due to a single causative variant, linkage will mean that there is uncertainty about its exact position.
As such, we lack a definitive rational basis to define when two QTL should be defined as mapping to the same place in the genome.
We have therefore followed previous studies [@agren_genetic_2013; @dittmar2014flowering; @oakley2014qtl; @agren_flowering_time; @postma2018among] in adopting a heuristic approach by defining QTL detected for different traits to colocalise, and hence be pleiotropic, if the confidence intervals of QTL positions overlapped.
This approach is practical, but such a simplification necessarily means that information about the subtleties of genetic architectures are lost.
For example, we were not able to assess the contribution of 'poorly-defined' QTL whose positions could not be resolved to within 15.2cM.
Such QTL might reflect either a single causative locus of weak effect, or multiple loci spread across the chromosome (figure \@ref(fig:qtl-pleiotropy)).
At the same time, we detected two QTL for seed mass in Sweden whose credible intervals were distinct, but which both fell into the single pleiotropic region Q10 (figure \@ref(fig:qtl-pleiotropy)).
These examples highlight that the pleiotropic regions should be interpreted with caution, and not seen as rigid boundaries separating pleiotropic and non-pleiotropic loci.

## No evidence for genetic constraint through seed mass

The Italian ecotype tended to produce somewhat larger seeds than did the Swedish ecotype, and this difference was particularly marked at the Italian site (supporting figure 1). Moreover, both ecotypes produced larger seeds at the field site in Sweden compared to that in Italy. Previous work in the same populations has demonstrated strong selection favouring the local ecotype during seedling establishment [@postma_early_2016]. In general, both probability of successful seedling establishment [@krannitz1991effect] and early growth rate [@el2004quantitative] should be positively related to seed size.
At the Italian site, the larger seeds of the Italian ecotype can thus be expected to contribute to a higher fitness of its offspring relative to that of the Swedish ecotype. However, the higher seedling establishment success of the local ecotype documented at the Swedish site [@postma_early_2016] cannot be explained by differences in seed size, but is more likely related to differences in seed dormancy and the timing of germination [cf. @postma2018among]. Further work is required to determine the  relative importance of seed size and other seed traits for differential seedling establishment in the two environments.

Although overlap between seed mass QTL and fecundity QTL indicated pleiotropic effects of several genomic regions (figure \@ref(fig:qtl-pleiotropy)), genetic correlations between seed size and fecundity in the RIL population tended to be weak (figure \@ref(fig:correlations). Seed mass and fecundity were negatively genetically correlated in Sweden in 2010, but no significant genetic correlation was detected in the other three site × year combinations. The overall weak genetic correlation between seed size and number can be explained by the fact that the direction of pleiotropic effects varied, and was positive roughly as frequently as negative. In contrast to theoretical predictions that there should be a trade-off between seed size and seed number [@Smith1974], these observations indicate that genetic correlations with seed mass place little constraint on the evolution of increased fecundity in these populations. One possible reason for the lack of a strong correlation between seed size and fecundity is that variation in seed size was too limited to affect fecundity and that trade-offs might be detected if genotypes with larger differences in seed size were crossed.

## Conclusions

In conclusion, this study has examined how variation in number of seeds per fruit and in resource allocation to different components of fitness contribute to overall adaptive differentiation in *A. thaliana*. Our results show that there is adaptive variation in seed production independent of variation in fruit number, and that the advantage to local genotypes can be underestimated if this is ignored. Moreover, we demonstrate consistent positive pleiotropy among components of fitness reflected in both genetic correlations among phenotypes and effects of underlying QTL, and very little evidence of a trade-off between offspring size and number. These findings indicate that the process of population divergence has been due in large part to the fixation of alleles that increase overall vigour or “condition” in different ways at each site.

# Acknowledgements

This work was made possible by the help of J. Glans, M. Vass, J. Trunschke, L. Vikström, F. Ågren, E. Chapurlat and F. Spada, and a large number of field assistants who helped with transplanting and harvesting field experiments. We also thank P. Falzini and Y. Jonsson for permission to conduct experiments on their land, and to the Botanical Garden of Rome for generously allowing us to use their greenhouse facilities.

# Funding

The study was financially supported by grants from the Swedish Research Council to JÅ and from the National Science Foundation (DEB 1743273) to CGO.

# Data availability

Data, R scripts, and the R markdown document used to create this manuscript will be uploaded to a suitable public server on publication. In the meantime, they are available at https://github.com/ellisztamas/fecundity_components.

# Competing interests

The authors declare no conflict of interest.

# Authors' contributions

TJE performed analyses and wrote the manuscript. JÅ conceived the study, and JÅ, FMP and CGO co-ordinated data collection and critically revised the manuscript.

# Literature cited

<div id="refs"></div>

# Supplementary material {-}

\beginsupplement

```{r individual-seed-qtl, fig.align='center', eval=T}
# number of seeds per fruit
ix <- "seed"
tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

for(i in 1:4){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  #tab[[i]]$PDE <- 100*abs(tab[[i]]$effect_sizes) / parent_diffs[parent_diffs$trait == ix, i]
}
tab <- do.call('rbind', tab)
# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
# print the table
kable(tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],#, "PDE")],
      format="pandoc",
      digits = c(0,0,0,0,0,1,3,1,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption="QTL for seeds/fruit identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects and as a percentage of variance among RIL means. Effect sizes are quantified as the least-square mean difference in seeds/fruit between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with fewer seeds/fruit.")
```

\newpage

```{r epistatic-qtl-table, eval=T}
# Traits to be presented.
traits <- c("frut","seed", "tofu", "tfit", 'mass')

epi_tab <- matrix(NA, nrow = 0, ncol=11)

for(tr in traits){
  for(sy in c("it2010", "it2011", "sw2010","sw2011")){
    eepi_tab <- grep(":", row.names(summary(qtl_fits[[tr]][[sy]])$result.drop))
    this_summ <- summary(qtl_fits[[tr]][[sy]])$result.drop[eepi_tab,]
    this_summ <- c(tr, substr(sy, 1,2), substr(sy, 3,6),
                   row.names(summary(qtl_fits[[tr]][[sy]])$result.drop)[eepi_tab], this_summ)
    if(!all(this_summ %in% c(tr, substr(sy, 1,2), substr(sy, 3,6)))) epi_tab <- rbind(epi_tab, this_summ)
  }
}

epi_tab <- as.data.frame(epi_tab)
colnames(epi_tab)[1:4] <- c("Trait", "Site","Year", "QTL")
epi_tab <- epi_tab[, c("Trait", "Site","Year", "QTL", "LOD", "%var")]
# Comvert values from factors
epi_tab$Trait  <- as.vector(epi_tab$Trait)
epi_tab$LOD    <- as.numeric(as.vector(epi_tab$LOD))
epi_tab$`%var` <- as.numeric(as.vector(epi_tab$`%var`))
# Make QTL labels look nicer
epi_split <- strsplit(as.character(epi_tab$QTL), ":")
epi_split <- sapply(epi_split, function(x) paste(x[1], "x", x[2]))
epi_tab$QTL <- epi_split

# Add prettier names
epi_tab$Trait[epi_tab$Trait == "mass"] <- "Seed mass"
epi_tab$Trait[epi_tab$Trait == "seed"] <- "Seeds/fruit"
epi_tab$Trait[epi_tab$Trait == "tofu"] <- "Seeds/RP"
epi_tab$Trait[epi_tab$Trait == "tfit"] <- "Seeds/seedling"
epi_tab$Trait[epi_tab$Trait == "frut"] <- "Fruits/RP"

epi_tab$Site <- ifelse(epi_tab$Site == "it", "Italy", "Sweden")

kable(epi_tab, digits = 2, row.names = F,
      format= "pandoc",
      col.names = c("Trait", "Site","Year", "QTL", "LOD", "% var. explained"),
      caption = "Pairs of significant epistatic QTL detected across traits.")
```

\newpage

```{r individual-ffit-qtl, eval=T}
# number of fruits per seedling
ix <- "ffit"
tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

for(i in 1:4){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$PDE <- 100*abs(tab[[i]]$effect_sizes) / parent_diffs[parent_diffs$trait == ix, i]
}
tab <- do.call('rbind', tab)
# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
# print the table
kable(tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],# "PDE")],
      format= "pandoc",
      digits = c(0,0,0,0,0,1,3,1,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption = "QTL for fruits/seedling identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects, and as a percentage of variance among RIL means. Effect sizes are quantified as the least-square mean difference in fruits/seedling between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with fewer fruits/seedling.")

```

\newpage

```{r individual-tfit-qtl, eval=T}
# seeds/seedling
ix <- "tfit"
tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

for(i in 1:4){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  #tab[[i]]$PDE <- 100*abs(tab[[i]]$effect_sizes) / parent_diffs[parent_diffs$trait == ix, i]
}
tab <- do.call('rbind', tab)
# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
# print the table
kable(tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],#, "PDE")],
      format= "pandoc",
      digits = c(0,0,0,0,0,1,3,1,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption = "QTL for seeds/seedling identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects and as a percentage of variance among RIL means. Effect sizes are quantified as the least-square mean difference in seeds/seedling between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with fewer seeds/seedling.")

```

\newpage

```{r individual-mass-qtl, eval=T}
# summary tables for each experiment
ix <- "mass"

tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

for(i in 1:4){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  #tab[[i]]$PDE <- 100*abs(tab[[i]]$effect_sizes) / parent_diffs[parent_diffs$trait == ix, i]
}
tab <- do.call('rbind', tab)
 
tab$site <- as.factor(tab$site)

# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
d <- tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")]
# print the table
kable(d,#tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],#, "PDE")],
      format= "pandoc",
      digits = c(0,0,0,0,0,1,3,1,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption="QTL for seed mass identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects and as a percentage of variance among RIL means. Effect sizes are quantified as the least-square mean difference in seed mass between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with lower seed mass.")
```

\newpage

```{r individual-surv-qtl, eval=T}
# survival to reproduction
site_names  <- c("Italy", "Italy", "Sweden")
year_names  <- c(2010, 2011, 2011)


ix <- "surv"
tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            #bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

pd <- parent_diffs[parent_diffs$trait == ix,c(1,2,4)]

for(i in 1:3){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$PDE <- (100* abs(tab[[i]]$effect_sizes)) / as.numeric(pd[i])
}
tab <- do.call('rbind', tab)
# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
# print the table
kable(tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],# "PDE")],
      format= "pandoc",
      digits = c(0,0,0,0,0,1,3,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption = "QTL for survival to reproduction identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects, as a percentage of variance among RIL means, and as a percentage of the difference between parental means. Effect sizes are quantified as the least-square mean difference in survival between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with lower survival.")

```

\newpage

```{r individual-frut-qtl, eval=T}
# summary tables for fruits/plant
ix <- "frut"
tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

for(i in 1:4){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  #tab[[i]]$PDE <- 100*abs(tab[[i]]$effect_sizes) / parent_diffs[parent_diffs$trait == ix, i]
}
tab <- do.call('rbind', tab)
# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
# print the table
kable(tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],#, "PDE")],
      format="pandoc",
      digits = c(0,0,0,0,0,1,3,1,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption="QTL for fruits/RP identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects and as a percentage of variance among RIL means. Effect sizes are quantified as the least-square mean difference in fruits/RP between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with fewer fruits/RP.")
```

\newpage

```{r individual-tofu-qtl, eval=T}
# total fecundity
ix <- "tofu"
tab <- list(bayesint_table(qtl_stepwise[[ix]]$it2010, qtl_fits[[ix]]$it2010),
            bayesint_table(qtl_stepwise[[ix]]$it2011, qtl_fits[[ix]]$it2011),
            bayesint_table(qtl_stepwise[[ix]]$sw2010, qtl_fits[[ix]]$sw2010),
            bayesint_table(qtl_stepwise[[ix]]$sw2011, qtl_fits[[ix]]$sw2011))

for(i in 1:4){
  tab[[i]]$site <- rep(site_names[i], nrow(tab[[i]])) # add experiment names
  tab[[i]]$year <- rep(year_names[i], nrow(tab[[i]])) # add experiment names
  #tab[[i]]$PDE <- 100*abs(tab[[i]]$effect_sizes) / parent_diffs[parent_diffs$trait == ix, i]
}
tab <- do.call('rbind', tab)
# mash together positions with their CIs
tab$position <- paste(format(round(tab$ML_bayesint,1), nsmall = 1),
                      " (",
                      format(round(tab$min_bayesint, 1), nsmall = 1), 
                      "-",
                      format(round(tab$max_bayesint, 1), nsmall = 1),
                      ")",
                      sep="")
# print the table
kable(tab[,c("name", "site", "year","chr","position","LOD", "effect_sizes","PVE")],#, "PDE")],
      format="pandoc",
      digits = c(0,0,0,0,0,1,3,1,1),
      col.names = c("Name", "Site", "Year","Chr.","Position (cM)","LOD", "Effect size","% var. explained"),# "% diff. explained"),
      caption = "QTL for seeds/RP identifed in individual experiments. QTL positions are shown with their 95% Bayesian credible intervals. Effect sizes are given as additive effects and as a percentage of variance among RIL means. Effect sizes are quantified as the least-square mean difference in seeds/RP between genotypes homozygous for the Swedish and Italian allele, respectively; negative values indicate that the Swedish genotype is associated with fewer seeds/RP.")

```

\newpage

```{r epistatic-seed, fig.width=16.9/2.54, fig.height=10/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for epistatic interactions detected for seeds/fruit, showing mean and standard errors for each genotype-by-genotype combination.", fig.scap = "Epistatic interactions for seeds/fruit", fig.align='center', eval=T}
# number of seeds per fruit
par(mfrow = c(1, 2))
effectplot(
  cross$seed,
  pheno.col = "sw2010",
  mname1 = "1@22.7",
  mname2 = "5@70.6",
  add.legend = F,
  geno2 = c("It. allele", "Sw. allele"),
  main = "Sweden 2010",
  xlab = "QTL 5@70.6",
  ylab = "Seeds/fruit"
)
legend(
  'bottomright',
  c("QTL 1@22.7", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex = 0.8
)

effectplot(
  cross$seed,
  pheno.col = "sw2011",
  mname1 = "1@61.1",
  mname2 = "3@21.0",
  add.legend = F,
  geno2 = c("It. allele", "Sw. allele"),
  main = "Sweden 2011",
  xlab = "QTL 3@21.0",
  ylab = "Seeds/fruit"
)
legend(
  'topright',
  c("QTL 1@61.1", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex = 0.8
)
```

\newpage

```{r seed-mass, fig.cap="Mean seed mass of the Italian (closed symbols) and Swedish (open symbols) parental ecotypes at the field sites in Italy (It.) and Sweden (Sw.) in 2010 and 2011. Bars indicate standard error.", fig.scap = "Seed mass among parental lines", fig.align= "center", fig.width=8/2.54}
# pdf("figures/02_seed_mass.pdf", width=11/2.54)

par(mfrow=c(1,1))
plot(c(1,4), c(15,30), type='n', axes = F, xlim=c(0.5, 4.5), xlab="",
     ylab=expression(paste("Seed mass (",mu, "g)")))
axis(2); box(); grid()
axis(1, 1:4, c("It. 2010", "It. 2011", "Sw. 2010", "Sw. 2011"), las=2, tick = T)

os <- 0.05
segments((1:4)-os, (parents$it_parent+parents$it_se)[parents$variable == 'seed_mass'],
         (1:4)-os, (parents$it_parent-parents$it_se)[parents$variable == 'seed_mass'],
         lwd=2)
segments((1:4)+os, (parents$sw_parent+parents$sw_se)[parents$variable == 'seed_mass'],
         (1:4)+os, (parents$sw_parent-parents$sw_se)[parents$variable == 'seed_mass'],
         lwd=2)
points((1:4)-os, parents$it_parent[parents$variable == 'seed_mass'], pch=16)
points((1:4)+os, parents$sw_parent[parents$variable == 'seed_mass'], pch=21, bg='white')


# dev.off()
```

\newpage

```{r epistatic-mass, fig.width=9/2.54, fig.height=10/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for epistatic interactions detected for seed mass, showing mean and standard errors for each genotype-by-genotype combination.", fig.scap = "Epistatic interactions for seed mass", fig.align='center', eval=T}
# seed mass
effectplot(
  cross$mass,
  pheno.col = "sw2011",
  mname1 = "3@55.4",
  mname2 = "5@78.2",
  add.legend = F,
  #ylim = c(500, 1100),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Sweden 2011",
  xlab = "QTL 5@78.2",
  ylab = "Seed mass"
)
legend('bottomright', c("QTL 3@55.4", "It. allele", "Sw. allele"), col = c(0,2,4), pch=1, bty="n", cex=0.8)

```

\newpage

```{r scatter-plots, fig.height=16/2.54, fig.width=16.9/2.54, fig.cap="Relationships between RIL-means for survival and fecundity (top), components of fecundity (middle), and between fecundity and seed mass (bottom). Pearson correlation coefficients ($r$), associated $p$-values and degrees of freedom (df) are indicated.", fig.scap = "Scatter plots between RIL-means for survival and fecundity, components of fecundity, and between overall fecundity and seed mass", fig.align= "center"}


corsumm <- data.frame(
  r = round(correlations$obs, 2),
  p = round(correlations$p, 3),
  df = correlations$df
)
corsumm$r <- paste("r = ", corsumm$r, sep="")
corsumm$p <- ifelse(correlations$p < 0.0001, "p < 0.0001", paste("p = ",corsumm$p, sep=""))
corsumm$df <- paste("df = ", corsumm$df, sep = "")

# setEPS()
# postscript(file = '../figures/trait_correlations.eps', family = "Times", width=16.9/2.56, height = 16/2.56)
# pdf(file = 'figures/03_genetic_correlations.pdf', width=16.9/2.56, height = 16/2.56)
par(mfrow=c(3,4), mar=c(4,3,1,0)+0.1, oma=c(0,1,1,1))

# fecundity vs survival
plot(ril_tofu$it2010, ril_surv$it2010, xlim=c(0,1600), ylim=c(0,1), xlab="", ylab="", main="Italy 2010")
# text(1550, 0.25, bquote(atop('r'[P] ==  .(corsumm$r[5]),  .(corsumm$tp[5]))), c(1,0))
legend(1300, -0.06, legend=corsumm[1,], bty='n', adj = 1, yjust = 0, xjust=0)

title(ylab="Survival to reproduction", line=2)
plot(ril_tofu$it2011, ril_surv$it2011, xlim=c(0,1000), ylim=c(0,1),  xlab="", ylab="", main="Italy 2011")
# text(1000, 0.45, bquote(atop('r'[P] ==  .(corsumm$r[6]),  .(corsumm$tp[6]))), c(1,0))
legend(800, -0.06, legend=corsumm[2,], bty='n', adj = 1, yjust = 0, xjust=0)

#
plot(ril_tofu$sw2010, ril_surv$sw2010,  xlim=c(0,5500), ylim=c(0, 1), xlab="", ylab="", main="Sweden 2010")
# text(5500, 0, bquote(atop('r'[P] ==  .(corsumm$r[7]),  .(corsumm$tp[7]))), c(1,0))
legend(4500,-0.06 , legend=corsumm[3,], bty='n', adj = 1, yjust = 0, xjust=0)

#
plot(ril_tofu$sw2011, ril_surv$sw2011,  xlim=c(0,4000), ylim=c(0,1), xlab="", ylab="", main="Sweden 2011")
# text(4000, 0, bquote(atop('r'[P] ==  .(corsumm$r[8]),  .(corsumm$tp[8]))), c(1,0))
legend(3200,-0.06 , legend=corsumm[4,], bty='n', adj = 1, yjust = 0, xjust=0)

title(xlab="Number of seeds per reproductive plant", outer=T, line=-32.7)


# Frut vs seed number
plot(ril_frut$it2010, ril_numb$it2010, xlim=c(0,40), ylim=c(0,50), xlab="", ylab="", pch=1)
# text(40, 10, bquote(atop('r'[P] ==  .(corsumm$r[1]),  .(corsumm$tp[1]))), c(1,0))
legend(32, -3, legend=corsumm[5,], bty='n', adj = 1, yjust = 0, xjust=0)
title(ylab="Number of seeds per fruit", line=2)
# 
plot(ril_frut$it2011, ril_numb$it2011, ylim=c(0,50), xlab="", ylab="", pch=1)
legend(21, -3 , legend=corsumm[6,], bty='n', adj = 1, yjust = 0, xjust=0)
# text(30, 10, bquote(atop('r'[P] ==  .(corsumm$r[2]),  .(corsumm$tp[2]))), c(1,0))
# 
plot(ril_frut$sw2010, ril_numb$sw2010, ylim=c(0,50), xlab="", ylab="", pch=1)
# text(100, 10, bquote(atop('r'[P] ==  .(corsumm$r[3]),  .(corsumm$tp[3]))), c(1,0))
legend(105, -3, legend=corsumm[7,], bty='n', adj = 1, yjust = 0, xjust=0)
# 
plot(ril_frut$sw2011, ril_numb$sw2011, ylim=c(0,50), xlab="", ylab="", pch=1)
legend(70, -3, legend=corsumm[8,], bty='n', adj = 1, yjust = 0, xjust=0)
# text(100, 10, bquote(atop('r'[P] ==  .(corsumm$r[4]),  .(corsumm$tp[4]))), c(1,0))
title(xlab="Number of fruits per reproductive plant", line=-17, outer=T)

# fecundity vs seed mass
plot(ril_tofu$it2010, ril_mass$it2010, xlim=c(0,1600), ylim=c(0,50), xlab="", ylab="")
# text(1550, 0, bquote(atop('r'[P] ==  .(corsumm$r[9]),  .(corsumm$tp[9]))), c(1,0))
legend(1300, -3 , legend=corsumm[9,], bty='n', adj = 1, yjust = 0, xjust=0)
title(ylab="Mean seed mass (\U00B5g)", line=2)
title(xlab="Number of seeds per reproductive plant", ylab="", outer=T, line=-1.5)

plot(ril_tofu$it2011, ril_mass$it2011, ylim=c(0,50), xlab="", ylab="")
legend(650, -3 , legend=corsumm[10,], bty='n', adj = 1, yjust = 0, xjust=0)
# text(1000, 0, bquote(atop('r'[P] ==  .(corsumm$r[10]),  .(corsumm$tp[10]))), c(1,0))

plot(ril_tofu$sw2010, ril_mass$sw2010, xlim=c(0,6000), ylim=c(0,50), xlab="", ylab="")
legend(4500, -3 , legend=corsumm[11,], bty='n', adj = 1, yjust = 0, xjust=0)
# text(6000, 0, bquote(atop('r'[P] ==  .(format(round(corsumm$r[11], 2), nsmall = 2)), .(corsumm$tp[11]))), c(1,0))

plot(ril_tofu$sw2011, ril_mass$sw2011, xlim=c(0,3500), ylim=c(0,50), xlab="", ylab="")
legend(2800, -3 , legend=corsumm[12,], bty='n', adj = 1, yjust = 0, xjust=0)
# text(3500, 0, bquote(atop('r'[P] ==  .(corsumm$r[12]),  .(corsumm$tp[12]))), c(1,0))
# # dev.off()
```

\newpage

```{r epistatic-frut, eval = T, fig.width = 16.9 / 2.54, fig.height = 10 /2.54, cache = T, cache.lazy = T, fig.cap = "Interaction plots for epistatic interactions detected for number of fruits per reproductive plant (fruits/RP), showing mean and standard errors for each genotype-by-genotype combination.", fig.scap = "Epistatic interactions for number of fruits per reproductive plant", fig.align='center', eval=T}
# number of seeds per plant
par(mfrow = c(1, 3))

effectplot(
  cross$frut,
  pheno.col = "it2010",
  mname1 = "5@7.5",
  mname2 = "5@70.1",
  add.legend = F,
  # ylim = c(000, 500),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Italy 2010",
  xlab = "QTL 5@70.1",
  ylab = "Fruits/RP"
)
legend(
  'bottomleft',
  c("QTL 5@7.5", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex=0.8
)

# number of seeds per plant
effectplot(
  cross$frut,
  pheno.col = "it2011",
  mname1 = "1@58.6",
  mname2 = "5@74.5",
  add.legend = F,
  # ylim = c(500, 1100),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Italy 2011",
  xlab = "QTL 5@74.5",
  ylab = "Fruits/RP"
)
legend(
  'bottomleft',
  c("QTL 1@58.6", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex=0.8
)

# number of seeds per plant
effectplot(
  cross$frut,
  pheno.col = "sw2011",
  mname1 = "1@61.8",
  mname2 = "3@26.7",
  add.legend = F,
  # ylim = c(500, 1100),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Sweden 2011",
  xlab = "QTL 3@26.7",
  ylab = "Seeds/RP"
)
legend(
  'bottomleft',
  c("QTL 1@61.8", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex=0.8
)


```

\newpage

```{r heritabilities, fig.cap="Broad-sense heritability estimates for survival to reproduction, fruits/RP, seeds/fruit (Seeds/fr), fruits/seedling (Fruits/sdl), and seed mass. Open and closed symbols show values in 2010 and 2011 respectively. Error bars show 95% parametric bootstrap confidence intervals. Numbers indicate the number of individuals each estimate is based on."}

# pdf("figures/heritabilities.pdf", width=16.9/2.54, height = 12/2.54)
par(mfrow=c(1,2))

os <- 0.1
plot(
  c(1, 6),
  c(0, 0.6),
  type = "n",
  axes = F,
  xlim = c(0.2, 5.5),
  xlab = "",
  ylab = "Broad-sense heritability",
  main = "Italy"
)
axis(1, 1:5, las=2,
     labels = c("Survival", "Fruits/RP", "Seeds/fr", "Fruits/sdl", "Seed mass"))
axis(2)
box()
# Confidence intervals
segments((1:5) - os, H2$lower$`Italy 2010`,
         (1:5) - os, H2$upper$`Italy 2010`)
segments((1:5) + os, H2$lower$`Italy 2011`,
         (1:5) + os, H2$upper$`Italy 2011`)
# Overplot observed values
points((1:5) - os, H2$obs$`Italy 2010`,   pch=16)
points((1:5) + os, H2$obs$`Italy 2011`,   pch=21, bg="white")

# Add sample sizes
text((1:5) - 1.5*os, 0.6,
     labels = H2$n$`Italy 2010`,
     adj = 1,
     srt=90,
     cex=0.7)
text((1:5) + 1.5*os, 0.6,
     labels = H2$n$`Italy 2011`,
     adj = 1,
     srt=90,
     cex=0.7)

# 2011
plot(c(1, 5), c(0,0.6), type="n", axes = F, xlim= c(0.5, 5.5),
     xlab="", ylab="Broad-sense heritability", main="Sweden")
axis(1, 1:5, las=2,
     labels = c("Survival", "Fruits/RP", "Seeds/fr", "Fruits/sdl", "Seed mass"))
axis(2)
box()
# Confidence intervals
segments((1:5) - os, H2$lower$`Sweden 2010`,
         (1:5) - os, H2$upper$`Sweden 2010`)
segments((1:5) + os, H2$lower$`Sweden 2011`,
         (1:5) + os, H2$upper$`Sweden 2011`)
# Overplot observed values
points((1:5) - os, H2$obs$`Sweden 2010`, pch=16)
points((1:5) + os, H2$obs$`Sweden 2011`, pch=21, bg="white")


# Add sample sizes
text((1:5) - 1.5*os, 0.6,
     labels = H2$n$`Sweden 2010`,
     srt=90,
     adj=1,
     cex=0.7)
text((1:5) + 1.5*os, 0.6,
     labels = H2$n$`Sweden 2011`,
     srt=90,
     adj=1,
     cex=0.7)

# dev.off()
```

\newpage

```{r epistatic-tofu, eval = T, fig.width = 16.9 / 2.54, fig.height = 10 /2.54, cache = T, cache.lazy = T, fig.cap = "Interaction plots for epistatic interactions detected for number of seeds per reproductive plant (seeds/RP), showing mean and standard errors for each genotype-by-genotype combination.", fig.scap = "Epistatic interactions for seeds/RP", fig.align='center', eval=T}
# number of seeds per plant
par(mfrow = c(1, 2))

effectplot(
  cross$tofu,
  pheno.col = "it2010",
  mname1 = "5@9.4",
  mname2 = "5@71.4",
  add.legend = F,
  ylim = c(000, 500),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Italy 2010",
  xlab = "QTL 5@71.4",
  ylab = "Seeds/RP"
)
legend(
  'bottomleft',
  c("QTL 5@9.4", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex=0.8
)

# number of seeds per plant
effectplot(
  cross$tofu,
  pheno.col = "sw2011",
  mname1 = "1@61.1",
  mname2 = "3@24.4",
  add.legend = F,
  ylim = c(500, 1100),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Sweden 2011",
  xlab = "QTL 3@24.4",
  ylab = "Seeds/RP"
)
legend(
  'bottomleft',
  c("QTL 1@61.1", "It. allele", "Sw. allele"),
  col = c(0, 2, 4),
  pch = 1,
  bty = "n",
  cex=0.8
)


```

\newpage

```{r epistatic-tfit, fig.width=16.9/2.54, fig.height=10/2.54, cache=T, cache.lazy=T, fig.cap="Interaction plots for epistatic interactions detected for number of seeds/seedling, showing mean and standard errors for each genotype-by-genotype combination.", fig.scap = "Epistatic interactions for number of seeds/seedling", fig.align='center', eval=T}
# number of seeds per plant
par(mfrow = c(1, 2))
effectplot(
  cross$tfit,
  pheno.col = "it2010",
  mname1 = "5@9.4",
  mname2 = "5@71.4",
  add.legend = F,
  #ylim = c(500, 1100),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Italy 2010",
  xlab = "QTL 5@71.4",
  ylab = "Seeds/seedling"
)
legend('bottomleft', c("QTL 5@9.4", "It. allele", "Sw. allele"), col = c(0,2,4), pch=1, bty="n", cex=0.8)

effectplot(
  cross$tfit,
  pheno.col = "sw2011",
  mname1 = "1@61.1",
  mname2 = "3@24.4",
  add.legend = F,
  #ylim = c(500, 1100),
  geno2 = c("It. allele", "Sw. allele"),
  main = "Sweden 2011",
  xlab = "QTL 3@24.4",
  ylab = "Seeds/seedling"
)
legend('bottomleft', c("QTL 1@61.1", "It. allele", "Sw. allele"), col = c(0,2,4), pch=1, bty="n", cex=0.8)

```


